---
title: "Memory Usage"
output: rmarkdown::html_vignette
self_contained: true
vignette: >
  %\VignetteIndexEntry{memory}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
options(rmarkdown.html_vignette.check_title = FALSE)
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  echo = FALSE,
  # eval = FALSE,
  message = FALSE,
  warning = FALSE
)
options(rmarkdown.html_vignette.check_title = FALSE)
```

```{r setup}
library(dplyr)
library(ggplot2)
library(glue)
library(readr)
library(MOSuite)
```

Dataset from: <https://portal.gdc.cancer.gov/projects/TCGA-LIHC>

```{r data}
counts <- read_tsv(system.file(
  "extdata", "LIHC_HTseqCounts.txt.gz",
  package = "MOSuite"
)) %>%
  rename(gene_id = Gene)
metadat <- read_tsv(system.file(
  "extdata", "LIHC_PatientData.txt.gz",
  package = "MOSuite"
)) %>%
  select(-sample_id) %>%
  rename(sample_id = barcode)

run_mosuite <- function(metadat, counts) {
  create_multiOmicDataSet_from_dataframes(metadat, counts) %>%
    clean_raw_counts() %>%
    calc_cpm() %>%
    filter_counts(
      group_colname = "treatments_radiation_treatment_type",
      label_colname = "sample"
    )
}
moo <- run_mosuite(metadat, counts)
```


```{r subset}
subset_mem <- lapply(c(10, 50, 100, 200, nrow(moo@sample_meta)), function(nsamples) {
  samples_subset <- moo@sample_meta %>% slice_sample(n = nsamples)
  counts_subset <- moo@counts$raw %>%
    dplyr::select(gene_id, tidyselect::all_of(samples_subset %>% dplyr::pull(sample_id)))

  moo_subset <- run_mosuite(samples_subset, counts_subset)
  return(tibble(
    n_samples = nsamples,
    object_size = lobstr::obj_size(moo_subset)
  ))
}) %>%
  bind_rows()
```

```{r single_cell}
to_bytes <- function(x, unit, base = 1024) {
  bytes_units <- list(
    K = 1,
    M = 2,
    G = 3,
    T = 4
  )
  return(x * (base^bytes_units[[unit]]))
}
sc_dat <- tibble::tribble(
  ~"project", ~"n_cells", ~"n_samples", ~"n_genes", ~"object_size_GB",
  "CCBR1329/CCBR1243", 64642, 10, 26359, 9.35,
  "CCBR1297", 9991, 2, 20989, 1.78,
  "CCBR1035", 170789, 19, 32100, 31.54,
  "CCBR1203", 208169, 35, 30858, 37.38
) %>%
  mutate(
    object_size = to_bytes(object_size_GB, "G", base = 1000),
    dataset_type = "single-cell (Seurat)"
  )
```


```{r plot_memory}
# what is the resource spec of default NIDAP allocation?
palette_name <- "Set2"
subset_mem %>%
  ggplot(aes(n_samples, object_size, color = "")) +
  geom_point() +
  geom_line(linewidth = 0.3) +
  scale_y_continuous(labels = scales::label_bytes(units = "GB", accuracy = 0.1)) +
  scale_color_manual(values = RColorBrewer::brewer.pal(3, palette_name)[[1]]) +
  labs(
    title = "Memory usage of S7 object",
    x = "Number of samples",
    y = "Memory usage",
    caption = glue(
      "The object contains sample metadata and count data as raw, CPM-transformed, and filtered counts.\n",
      "Each dataset has the same number of genes: ",
      format(length(moo@counts$raw$gene_id), big.mark = ",")
    )
  ) +
  theme_bw() +
  theme(legend.position = "none")
```

```{r plot_memory_comp}
dat_comp <- bind_rows(
  subset_mem %>% mutate(dataset_type = "bulk (S7)", object_size = as.double(object_size)),
  sc_dat
)
dat_comp %>%
  ggplot(aes(n_samples, object_size, colour = dataset_type)) +
  geom_point() +
  geom_line(linewidth = 0.3) +
  scale_y_continuous(labels = scales::label_bytes(units = "GB", accuracy = 0.1)) +
  scale_color_brewer(palette = palette_name) +
  guides(colour = guide_legend(
    title = "",
    position = "top",
    reverse = TRUE,
    theme = theme(legend.text.position = "top")
  )) +
  labs(title = "Memory usage of bulk & single-cell RNA-seq objects", x = "Number of samples", y = "Memory usage") +
  theme_bw()
```
