<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to MultiOmicsSuite • MOSuite</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to MultiOmicsSuite">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.2.1.9001</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
    <li><a class="dropdown-item" href="../articles/cli.html">Calling MOSuite from the CLI</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to MultiOmicsSuite</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/vignettes/intro.Rmd" class="external-link"><code>vignettes/intro.Rmd</code></a></small>
      <div class="d-none name"><code>intro.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CCBR/MOSuite" class="external-link">MOSuite</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/options.html">options</a></span><span class="op">(</span>moo_print_plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">moo_nidap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_multiOmicDataSet_from_dataframes.html">create_multiOmicDataSet_from_dataframes</a></span><span class="op">(</span></span>
<span>  sample_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_sample_metadata</span><span class="op">)</span>,</span>
<span>  counts_dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_raw_counts</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/clean_raw_counts.html">clean_raw_counts</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_counts.html">filter_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/batch_correct_counts.html">batch_correct_counts</a></span><span class="op">(</span></span>
<span>    covariates_colname <span class="op">=</span> <span class="st">"Group"</span>,</span>
<span>    batch_colname <span class="op">=</span> <span class="st">"Batch"</span>,</span>
<span>    label_colname <span class="op">=</span> <span class="st">"Label"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/diff_counts.html">diff_counts</a></span><span class="op">(</span></span>
<span>    count_type <span class="op">=</span> <span class="st">"filt"</span>,</span>
<span>    covariates_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span>
<span>    contrast_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A"</span>, <span class="st">"C-A"</span>, <span class="st">"B-C"</span><span class="op">)</span>,</span>
<span>    input_in_log_counts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_mean_and_sd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    voom_normalization_method <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_diff.html">filter_diff</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="intro_files/figure-html/nidap_data-1.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; * cleaning raw counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Not able to identify multiple id's in GeneName</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Columns that can be used to aggregate gene information GeneName</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Aggregating the counts for the same ID in different chromosome locations.</span></span>
<span><span class="co">#&gt; Column used to Aggregate duplicate IDs: GeneName</span></span>
<span><span class="co">#&gt; Number of rows before Collapse: 43280</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; no duplicated IDs in GeneName</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; * filtering clean counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of features after filtering: 7943</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; colors_for_plots NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; colors_for_plots character</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-2.png" class="r-plt" alt="" width="700"><img src="intro_files/figure-html/nidap_data-3.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-4.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; * normalizing filt counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Total number of features included: 7943</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-5.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-6.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-7.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; Sample columns: A1, Sample columns: A2, Sample columns: A3, Sample columns: B1, Sample columns: B2, Sample columns: B3, Sample columns: C1, Sample columns: C2, Sample columns: C3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; * batch-correcting norm-voom counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Found2batches</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Adjusting for2covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Finding parametric adjustments</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-8.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-9.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-10.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; The total number of features in output: 7943</span></span>
<span><span class="co">#&gt; Number of samples after batch correction: 10</span></span>
<span><span class="co">#&gt; * differential counts</span></span>
<span><span class="co">#&gt; Setting first column of `counts` as gene annotation.</span></span>
<span><span class="co">#&gt; Total number of genes included: 7943</span></span>
<span><span class="co">#&gt; `geom_smooth()` using method = 'gam' and formula = 'y ~ s(x, bs = "cs")'</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-11.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; `geom_smooth()` using method = 'gam' and formula = 'y ~ s(x, bs = "cs")'</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; * filtering differential features</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Total number of genes selected with adjpval &lt; 0.05 and | logFC | ≥ 1 is sum(selgenes)</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-12.png" class="r-plt" alt="" width="700"></p>
<pre><code><span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span></span>
<span><span class="va">moo_nidap</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/join_dfs_wide.html">join_dfs_wide</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt;   GeneName   B-A_FC  B-A_logFC B-A_tstat  B-A_pval B-A_adjpval    C-A_FC</span></span>
<span><span class="co">#&gt; 1   Mrpl15 1.059250 0.08304265 0.2377167 0.8162052   0.9682636 -1.068725</span></span>
<span><span class="co">#&gt; 2   Lypla1 1.370909 0.45513310 1.1301522 0.2810081   0.7797612 -1.066981</span></span>
<span><span class="co">#&gt; 3    Tcea1 1.083699 0.11596469 0.3657617 0.7210585   0.9500109 -1.177051</span></span>
<span><span class="co">#&gt; 4  Atp6v1h 1.311199 0.39088683 1.1241780 0.2834330   0.7825194 -1.221374</span></span>
<span><span class="co">#&gt; 5   Rb1cc1 1.514888 0.59921070 1.3095182 0.2154459   0.7187843  1.313927</span></span>
<span><span class="co">#&gt; 6   Pcmtd1 1.112738 0.15411405 0.2497788 0.8070821   0.9663212  3.238362</span></span>
<span><span class="co">#&gt;     C-A_logFC  C-A_tstat    C-A_pval C-A_adjpval    B-C_FC  B-C_logFC</span></span>
<span><span class="co">#&gt; 1 -0.09589040 -0.2897955 0.777035116  0.89204033  1.132046  0.1789331</span></span>
<span><span class="co">#&gt; 2 -0.09353458 -0.2379598 0.816021040  0.91228196  1.462734  0.5486677</span></span>
<span><span class="co">#&gt; 3 -0.23517638 -0.7602905 0.462100568  0.69160897  1.275569  0.3511411</span></span>
<span><span class="co">#&gt; 4 -0.28850521 -0.8324070 0.421815393  0.65837683  1.601465  0.6793920</span></span>
<span><span class="co">#&gt; 5  0.39388567  0.9073581 0.382492765  0.62547476  1.152946  0.2053250</span></span>
<span><span class="co">#&gt; 6  1.69526432  3.4021010 0.005417489  0.05722644 -2.910264 -1.5411503</span></span>
<span><span class="co">#&gt;    B-C_tstat   B-C_pval B-C_adjpval</span></span>
<span><span class="co">#&gt; 1  0.5285947 0.60695058   0.8355016</span></span>
<span><span class="co">#&gt; 2  1.3910079 0.19006353   0.4871490</span></span>
<span><span class="co">#&gt; 3  1.1281746 0.28180907   0.5893629</span></span>
<span><span class="co">#&gt; 4  1.9865449 0.07085778   0.3168579</span></span>
<span><span class="co">#&gt; 5  0.4843119 0.63708764   0.8518020</span></span>
<span><span class="co">#&gt; 6 -2.9536960 0.01233497   0.1411803</span></span>
<span></span>
<span><span class="va">moo_nidap</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff_filt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   GeneName B-A_FC B-A_logFC B-A_tstat B-A_pval B-A_adjpval C-A_FC C-A_logFC</span></span>
<span><span class="co">#&gt; 1     Rrs1  -2.06    -1.040    -2.860   0.0147       0.276  -2.71     -1.44</span></span>
<span><span class="co">#&gt; 2     Mcm3  -1.45    -0.539    -1.870   0.0869       0.544  -2.45     -1.29</span></span>
<span><span class="co">#&gt; 3   Ogfrl1   1.07     0.102     0.292   0.7750       0.960  -3.78     -1.92</span></span>
<span><span class="co">#&gt; 4    Smap1   2.96     1.570     2.010   0.0684       0.499   5.68      2.51</span></span>
<span><span class="co">#&gt; 5  Plekhb2  -1.24    -0.312    -1.100   0.2950       0.789   2.69      1.43</span></span>
<span><span class="co">#&gt; 6   Il18r1   2.41     1.270     0.706   0.4940       0.875  36.40      5.19</span></span>
<span><span class="co">#&gt;   C-A_tstat C-A_pval C-A_adjpval B-C_FC B-C_logFC B-C_tstat B-C_pval</span></span>
<span><span class="co">#&gt; 1     -3.83 2.48e-03     0.03680   1.32     0.399     0.941 3.66e-01</span></span>
<span><span class="co">#&gt; 2     -4.33 1.03e-03     0.02240   1.69     0.755     2.380 3.53e-02</span></span>
<span><span class="co">#&gt; 3     -4.04 1.71e-03     0.03030   4.05     2.020     4.030 1.76e-03</span></span>
<span><span class="co">#&gt; 4      3.59 3.82e-03     0.04700  -1.92    -0.938    -1.750 1.07e-01</span></span>
<span><span class="co">#&gt; 5      5.99 6.94e-05     0.00463  -3.34    -1.740    -6.980 1.66e-05</span></span>
<span><span class="co">#&gt; 6      3.13 8.86e-03     0.07640 -15.10    -3.920    -4.230 1.23e-03</span></span>
<span><span class="co">#&gt;   B-C_adjpval</span></span>
<span><span class="co">#&gt; 1      0.6660</span></span>
<span><span class="co">#&gt; 2      0.2300</span></span>
<span><span class="co">#&gt; 3      0.0550</span></span>
<span><span class="co">#&gt; 4      0.3780</span></span>
<span><span class="co">#&gt; 5      0.0033</span></span>
<span><span class="co">#&gt; 6      0.0455</span></span></code></pre>
<div class="section level2">
<h2 id="the-multiomicdataset-object-structure">The multiOmicDataSet object structure<a class="anchor" aria-label="anchor" href="#the-multiomicdataset-object-structure"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">moo_nidap</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt;MOSuite::multiOmicDataSet&gt;</span></span>
<span><span class="co">#&gt;  @ sample_meta:'data.frame': 9 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;  .. $ Sample   : chr  "A1" "A2" "A3" "B1" ...</span></span>
<span><span class="co">#&gt;  .. $ Group    : chr  "A" "A" "A" "B" ...</span></span>
<span><span class="co">#&gt;  .. $ Replicate: num  1 2 3 1 2 3 1 2 3</span></span>
<span><span class="co">#&gt;  .. $ Batch    : num  1 2 2 1 1 2 1 2 2</span></span>
<span><span class="co">#&gt;  .. $ Label    : chr  "A1" "A2" "A3" "B1" ...</span></span>
<span><span class="co">#&gt;  .. - attr(*, "spec")=List of 3</span></span>
<span><span class="co">#&gt;  ..  ..$ cols   :List of 5</span></span>
<span><span class="co">#&gt;  ..  .. ..$ Sample   : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ Group    : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ Replicate: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ Batch    : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ Label    : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  ..$ default: list()</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "class")= chr [1:2] "collector_guess" "collector"</span></span>
<span><span class="co">#&gt;  ..  ..$ delim  : chr ","</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "class")= chr "col_spec"</span></span>
<span><span class="co">#&gt;  .. - attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  @ annotation :'data.frame': 43280 obs. of  1 variable:</span></span>
<span><span class="co">#&gt;  .. $ GeneName: chr  "RP23-271O17.1" "Gm26206" "Xkr4" "RP23-317L18.1" ...</span></span>
<span><span class="co">#&gt;  .. - attr(*, "spec")=List of 3</span></span>
<span><span class="co">#&gt;  ..  ..$ cols   :List of 10</span></span>
<span><span class="co">#&gt;  ..  .. ..$ GeneName: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  ..$ default: list()</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "class")= chr [1:2] "collector_guess" "collector"</span></span>
<span><span class="co">#&gt;  ..  ..$ delim  : chr ","</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "class")= chr "col_spec"</span></span>
<span><span class="co">#&gt;  .. - attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  @ counts     :List of 5</span></span>
<span><span class="co">#&gt;  .. $ raw  :'data.frame':    43280 obs. of  10 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ GeneName: chr [1:43280] "RP23-271O17.1" "Gm26206" "Xkr4" "RP23-317L18.1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A2      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B2      : num [1:43280] 0 0 0 0 0 0 0 0 0 1 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C2      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "spec")=List of 3</span></span>
<span><span class="co">#&gt;  ..  .. ..$ cols   :List of 10</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ GeneName: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ default: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_guess" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ delim  : chr ","</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "class")= chr "col_spec"</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  .. $ clean:'data.frame':    43280 obs. of  10 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ GeneName: chr [1:43280] "RP23-271O17.1" "Gm26206" "Xkr4" "RP23-317L18.1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A2      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B2      : num [1:43280] 0 0 0 0 0 0 0 0 0 1 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C1      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C2      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C3      : num [1:43280] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "spec")=List of 3</span></span>
<span><span class="co">#&gt;  ..  .. ..$ cols   :List of 10</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ GeneName: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ default: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_guess" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ delim  : chr ","</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "class")= chr "col_spec"</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  .. $ filt :'data.frame':    7943 obs. of  10 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A1      : num [1:7943] 1245 1483 1381 1033 666 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A2      : num [1:7943] 1341 1410 2044 1959 1397 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A3      : num [1:7943] 1476 1370 2051 1890 1576 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B1      : num [1:7943] 965 1146 2325 2075 681 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B2      : num [1:7943] 1235 1422 2386 2702 2040 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B3      : num [1:7943] 1784 2624 1893 2150 1988 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C1      : num [1:7943] 1058 991 2391 2436 774 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C2      : num [1:7943] 1732 1101 916 1321 1921 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C3      : num [1:7943] 1531 2352 2261 1018 2660 ...</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "spec")=List of 3</span></span>
<span><span class="co">#&gt;  ..  .. ..$ cols   :List of 10</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ GeneName: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_character" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ A3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ B3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C1      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C2      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. .. ..$ C3      : list()</span></span>
<span><span class="co">#&gt;  ..  .. .. .. ..- attr(*, "class")= chr [1:2] "collector_double" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ default: list()</span></span>
<span><span class="co">#&gt;  ..  .. .. ..- attr(*, "class")= chr [1:2] "collector_guess" "collector"</span></span>
<span><span class="co">#&gt;  ..  .. ..$ delim  : chr ","</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "class")= chr "col_spec"</span></span>
<span><span class="co">#&gt;  ..  ..- attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  .. $ norm :List of 1</span></span>
<span><span class="co">#&gt;  ..  ..$ voom:'data.frame':  7943 obs. of  10 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A1      : num [1:7943] 6.79 7.04 6.93 6.51 5.85 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A2      : num [1:7943] 6.7 6.78 7.32 7.26 6.76 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ A3      : num [1:7943] 6.64 6.53 7.13 7 6.75 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B1      : num [1:7943] 6.22 6.47 7.48 7.31 5.75 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B2      : num [1:7943] 6.38 6.57 7.33 7.52 7.11 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ B3      : num [1:7943] 7.02 7.59 7.11 7.3 7.17 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C1      : num [1:7943] 6.02 5.92 7.19 7.22 5.56 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C2      : num [1:7943] 7.03 6.43 6.19 6.67 7.19 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ C3      : num [1:7943] 6.6 7.25 7.2 6.01 7.43 ...</span></span>
<span><span class="co">#&gt;  .. $ batch:'data.frame':    7943 obs. of  10 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A1      : num [1:7943] 6.92 7.2 6.87 6.51 6.23 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A2      : num [1:7943] 6.59 6.66 7.33 7.25 6.46 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ A3      : num [1:7943] 6.54 6.43 7.16 7.02 6.45 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B1      : num [1:7943] 6.41 6.68 7.42 7.28 6.27 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B2      : num [1:7943] 6.55 6.78 7.27 7.47 7.35 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B3      : num [1:7943] 6.88 7.4 7.16 7.33 6.89 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C1      : num [1:7943] 6.21 6.13 7.13 7.13 6.06 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C2      : num [1:7943] 6.87 6.32 6.32 6.69 6.88 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C3      : num [1:7943] 6.48 7.04 7.18 6.1 7.1 ...</span></span>
<span><span class="co">#&gt;  @ analyses   :List of 3</span></span>
<span><span class="co">#&gt;  .. $ colors   :List of 5</span></span>
<span><span class="co">#&gt;  ..  ..$ Sample   : Named chr [1:9] "#000000" "#E69F00" "#56B4E9" "#009E73" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:9] "A1" "A2" "A3" "B1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ Group    : Named chr [1:3] "#000000" "#E69F00" "#56B4E9"</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:3] "A" "B" "C"</span></span>
<span><span class="co">#&gt;  ..  ..$ Replicate: Named chr [1:3] "#000000" "#E69F00" "#56B4E9"</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:3] "1" "2" "3"</span></span>
<span><span class="co">#&gt;  ..  ..$ Batch    : Named chr [1:2] "#000000" "#E69F00"</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:2] "1" "2"</span></span>
<span><span class="co">#&gt;  ..  ..$ Label    : Named chr [1:9] "#000000" "#E69F00" "#56B4E9" "#009E73" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:9] "A1" "A2" "A3" "B1" ...</span></span>
<span><span class="co">#&gt;  .. $ diff     :List of 3</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A:'data.frame':   7943 obs. of  6 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ FC      : num [1:7943] 1.06 1.37 1.08 1.31 1.51 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ logFC   : num [1:7943] 0.083 0.455 0.116 0.391 0.599 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ tstat   : num [1:7943] 0.238 1.13 0.366 1.124 1.31 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ pval    : num [1:7943] 0.816 0.281 0.721 0.283 0.215 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ adjpval : num [1:7943] 0.968 0.78 0.95 0.783 0.719 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A:'data.frame':   7943 obs. of  6 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ FC      : num [1:7943] -1.07 -1.07 -1.18 -1.22 1.31 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ logFC   : num [1:7943] -0.0959 -0.0935 -0.2352 -0.2885 0.3939 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ tstat   : num [1:7943] -0.29 -0.238 -0.76 -0.832 0.907 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ pval    : num [1:7943] 0.777 0.816 0.462 0.422 0.382 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ adjpval : num [1:7943] 0.892 0.912 0.692 0.658 0.625 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C:'data.frame':   7943 obs. of  6 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ GeneName: chr [1:7943] "Mrpl15" "Lypla1" "Tcea1" "Atp6v1h" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ FC      : num [1:7943] 1.13 1.46 1.28 1.6 1.15 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ logFC   : num [1:7943] 0.179 0.549 0.351 0.679 0.205 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ tstat   : num [1:7943] 0.529 1.391 1.128 1.987 0.484 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ pval    : num [1:7943] 0.607 0.1901 0.2818 0.0709 0.6371 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ adjpval : num [1:7943] 0.836 0.487 0.589 0.317 0.852 ...</span></span>
<span><span class="co">#&gt;  .. $ diff_filt:'data.frame':    635 obs. of  16 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ GeneName   : chr [1:635] "Rrs1" "Mcm3" "Ogfrl1" "Smap1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A_FC     : num [1:635] -2.06 -1.45 1.07 2.96 -1.24 2.41 -1.63 2.41 53.5 1.63 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A_logFC  : num [1:635] -1.04 -0.539 0.102 1.57 -0.312 1.27 -0.706 1.27 5.74 0.708 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A_tstat  : num [1:635] -2.86 -1.87 0.292 2.01 -1.1 0.706 -1.9 3.2 3.64 1.91 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A_pval   : num [1:635] 0.0147 0.0869 0.775 0.0684 0.295 0.494 0.0827 0.00784 0.00354 0.0804 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-A_adjpval: num [1:635] 0.276 0.544 0.96 0.499 0.789 0.875 0.535 0.21 0.142 0.53 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A_FC     : num [1:635] -2.71 -2.45 -3.78 5.68 2.69 36.4 3.13 2.74 287 -6.39 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A_logFC  : num [1:635] -1.44 -1.29 -1.92 2.51 1.43 5.19 1.64 1.46 8.17 -2.68 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A_tstat  : num [1:635] -3.83 -4.33 -4.04 3.59 5.99 3.13 5.63 3.97 5.42 -3.83 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A_pval   : num [1:635] 2.48e-03 1.03e-03 1.71e-03 3.82e-03 6.94e-05 8.86e-03 1.22e-04 1.94e-03 1.68e-04 2.50e-03 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ C-A_adjpval: num [1:635] 0.0368 0.0224 0.0303 0.047 0.00463 0.0764 0.00644 0.0322 0.00784 0.0369 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C_FC     : num [1:635] 1.32 1.69 4.05 -1.92 -3.34 -15.1 -5.1 -1.14 -5.37 10.4 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C_logFC  : num [1:635] 0.399 0.755 2.02 -0.938 -1.74 -3.92 -2.35 -0.189 -2.42 3.38 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C_tstat  : num [1:635] 0.941 2.38 4.03 -1.75 -6.98 -4.23 -7.09 -0.583 -4.01 4.57 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C_pval   : num [1:635] 3.66e-01 3.53e-02 1.76e-03 1.07e-01 1.66e-05 1.23e-03 1.44e-05 5.71e-01 1.82e-03 6.84e-04 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ B-C_adjpval: num [1:635] 0.666 0.23 0.055 0.378 0.0033 0.0455 0.00309 0.816 0.0562 0.0333 ...</span></span></code></pre></div>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
