<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to MultiOmicsSuite • MOSuite</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to MultiOmicsSuite">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/cli.html">Calling MOSuite from the CLI</a></li>
    <li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article" id="container">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to MultiOmicsSuite</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/vignettes/intro.Rmd" class="external-link"><code>vignettes/intro.Rmd</code></a></small>
      <div class="d-none name"><code>intro.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CCBR/MOSuite" class="external-link">MOSuite</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/options.html">options</a></span><span class="op">(</span>moo_print_plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">moo_nidap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_multiOmicDataSet_from_dataframes.html">create_multiOmicDataSet_from_dataframes</a></span><span class="op">(</span></span>
<span>  sample_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_sample_metadata</span><span class="op">)</span>,</span>
<span>  counts_dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_raw_counts</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/clean_raw_counts.html">clean_raw_counts</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_counts.html">filter_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/batch_correct_counts.html">batch_correct_counts</a></span><span class="op">(</span></span>
<span>    covariates_colname <span class="op">=</span> <span class="st">"Group"</span>,</span>
<span>    batch_colname <span class="op">=</span> <span class="st">"Batch"</span>,</span>
<span>    label_colname <span class="op">=</span> <span class="st">"Label"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/diff_counts.html">diff_counts</a></span><span class="op">(</span></span>
<span>    count_type <span class="op">=</span> <span class="st">"filt"</span>,</span>
<span>    covariates_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span>
<span>    contrast_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A"</span>, <span class="st">"C-A"</span>, <span class="st">"B-C"</span><span class="op">)</span>,</span>
<span>    input_in_log_counts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_mean_and_sd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    voom_normalization_method <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_diff.html">filter_diff</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="intro_files/figure-html/nidap_data-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; * cleaning raw counts</span></span>
<span><span class="co">#&gt; Not able to identify multiple id's in GeneName</span></span>
<span><span class="co">#&gt; Columns that can be used to aggregate gene information GeneName</span></span>
<span><span class="co">#&gt; Aggregating the counts for the same ID in different chromosome locations.</span></span>
<span><span class="co">#&gt; Column used to Aggregate duplicate IDs: GeneName</span></span>
<span><span class="co">#&gt; Number of rows before Collapse: 43280</span></span>
<span><span class="co">#&gt; no duplicated IDs in GeneName</span></span>
<span><span class="co">#&gt; * filtering clean counts</span></span>
<span><span class="co">#&gt; Number of features after filtering: 7943</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-2.png" width="700"><img src="intro_files/figure-html/nidap_data-3.png" width="700"><img src="intro_files/figure-html/nidap_data-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; * normalizing filt counts</span></span>
<span><span class="co">#&gt; Total number of features included: 7943</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-5.png" width="700"><img src="intro_files/figure-html/nidap_data-6.png" width="700"><img src="intro_files/figure-html/nidap_data-7.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Sample columns: A1, Sample columns: A2, Sample columns: A3, Sample columns: B1, Sample columns: B2, Sample columns: B3, Sample columns: C1, Sample columns: C2, Sample columns: C3</span></span>
<span><span class="co">#&gt; * batch-correcting norm-voom counts</span></span>
<span><span class="co">#&gt; Found2batches</span></span>
<span><span class="co">#&gt; Adjusting for2covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; Finding parametric adjustments</span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-8.png" width="700"><img src="intro_files/figure-html/nidap_data-9.png" width="700"><img src="intro_files/figure-html/nidap_data-10.png" width="700"></p>
<pre><code><span><span class="co">#&gt; The total number of features in output: 7943</span></span>
<span><span class="co">#&gt; Number of samples after batch correction: 10</span></span>
<span><span class="co">#&gt; * differential counts</span></span>
<span><span class="co">#&gt; Setting first column of `counts` as gene annotation.</span></span>
<span><span class="co">#&gt; Total number of genes included: 7943</span></span>
<span><span class="co">#&gt; `geom_smooth()` using method = 'gam' and formula = 'y ~ s(x, bs = "cs")'</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-11.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; * filtering differential features</span></span>
<span><span class="co">#&gt; Total number of genes selected with adjpval &lt; 0.05 and | logFC | ≥ 1 is</span></span>
<span><span class="co">#&gt; sum(selgenes)</span></span></code></pre>
<p><img src="intro_files/figure-html/nidap_data-12.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">moo_nidap</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/join_dfs_wide.html">join_dfs_wide</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt;   GeneName   B-A_FC  B-A_logFC B-A_tstat  B-A_pval B-A_adjpval    C-A_FC</span></span>
<span><span class="co">#&gt; 1   Mrpl15 1.059250 0.08304265 0.2377167 0.8162052   0.9682636 -1.068725</span></span>
<span><span class="co">#&gt; 2   Lypla1 1.370909 0.45513310 1.1301522 0.2810081   0.7797612 -1.066981</span></span>
<span><span class="co">#&gt; 3    Tcea1 1.083699 0.11596469 0.3657617 0.7210585   0.9500109 -1.177051</span></span>
<span><span class="co">#&gt; 4  Atp6v1h 1.311199 0.39088683 1.1241780 0.2834330   0.7825194 -1.221374</span></span>
<span><span class="co">#&gt; 5   Rb1cc1 1.514888 0.59921070 1.3095182 0.2154459   0.7187843  1.313927</span></span>
<span><span class="co">#&gt; 6   Pcmtd1 1.112738 0.15411405 0.2497788 0.8070821   0.9663212  3.238362</span></span>
<span><span class="co">#&gt;     C-A_logFC  C-A_tstat    C-A_pval C-A_adjpval    B-C_FC  B-C_logFC</span></span>
<span><span class="co">#&gt; 1 -0.09589040 -0.2897955 0.777035116  0.89204033  1.132046  0.1789331</span></span>
<span><span class="co">#&gt; 2 -0.09353458 -0.2379598 0.816021040  0.91228196  1.462734  0.5486677</span></span>
<span><span class="co">#&gt; 3 -0.23517638 -0.7602905 0.462100568  0.69160897  1.275569  0.3511411</span></span>
<span><span class="co">#&gt; 4 -0.28850521 -0.8324070 0.421815393  0.65837683  1.601465  0.6793920</span></span>
<span><span class="co">#&gt; 5  0.39388567  0.9073581 0.382492765  0.62547476  1.152946  0.2053250</span></span>
<span><span class="co">#&gt; 6  1.69526432  3.4021010 0.005417489  0.05722644 -2.910264 -1.5411503</span></span>
<span><span class="co">#&gt;    B-C_tstat   B-C_pval B-C_adjpval</span></span>
<span><span class="co">#&gt; 1  0.5285947 0.60695058   0.8355016</span></span>
<span><span class="co">#&gt; 2  1.3910079 0.19006353   0.4871490</span></span>
<span><span class="co">#&gt; 3  1.1281746 0.28180907   0.5893629</span></span>
<span><span class="co">#&gt; 4  1.9865449 0.07085778   0.3168579</span></span>
<span><span class="co">#&gt; 5  0.4843119 0.63708764   0.8518020</span></span>
<span><span class="co">#&gt; 6 -2.9536960 0.01233497   0.1411803</span></span>
<span></span>
<span><span class="va">moo_nidap</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff_filt</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   GeneName B-A_FC B-A_logFC B-A_tstat B-A_pval B-A_adjpval C-A_FC C-A_logFC</span></span>
<span><span class="co">#&gt; 1     Rrs1  -2.06    -1.040    -2.860   0.0147       0.276  -2.71     -1.44</span></span>
<span><span class="co">#&gt; 2     Mcm3  -1.45    -0.539    -1.870   0.0869       0.544  -2.45     -1.29</span></span>
<span><span class="co">#&gt; 3   Ogfrl1   1.07     0.102     0.292   0.7750       0.960  -3.78     -1.92</span></span>
<span><span class="co">#&gt; 4    Smap1   2.96     1.570     2.010   0.0684       0.499   5.68      2.51</span></span>
<span><span class="co">#&gt; 5  Plekhb2  -1.24    -0.312    -1.100   0.2950       0.789   2.69      1.43</span></span>
<span><span class="co">#&gt; 6   Il18r1   2.41     1.270     0.706   0.4940       0.875  36.40      5.19</span></span>
<span><span class="co">#&gt;   C-A_tstat C-A_pval C-A_adjpval B-C_FC B-C_logFC B-C_tstat B-C_pval</span></span>
<span><span class="co">#&gt; 1     -3.83 2.48e-03     0.03680   1.32     0.399     0.941 3.66e-01</span></span>
<span><span class="co">#&gt; 2     -4.33 1.03e-03     0.02240   1.69     0.755     2.380 3.53e-02</span></span>
<span><span class="co">#&gt; 3     -4.04 1.71e-03     0.03030   4.05     2.020     4.030 1.76e-03</span></span>
<span><span class="co">#&gt; 4      3.59 3.82e-03     0.04700  -1.92    -0.938    -1.750 1.07e-01</span></span>
<span><span class="co">#&gt; 5      5.99 6.94e-05     0.00463  -3.34    -1.740    -6.980 1.66e-05</span></span>
<span><span class="co">#&gt; 6      3.13 8.86e-03     0.07640 -15.10    -3.920    -4.230 1.23e-03</span></span>
<span><span class="co">#&gt;   B-C_adjpval</span></span>
<span><span class="co">#&gt; 1      0.6660</span></span>
<span><span class="co">#&gt; 2      0.2300</span></span>
<span><span class="co">#&gt; 3      0.0550</span></span>
<span><span class="co">#&gt; 4      0.3780</span></span>
<span><span class="co">#&gt; 5      0.0033</span></span>
<span><span class="co">#&gt; 6      0.0455</span></span></code></pre></div>
  </main>
</div>



   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer>
</body>
</html>
