<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>RSEM counts from RENEE • MOSuite</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="RSEM counts from RENEE">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.2.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
    <li><a class="dropdown-item" href="../articles/cli.html">Calling MOSuite from the CLI</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article" id="container">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>RSEM counts from RENEE</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/vignettes/renee.Rmd" class="external-link"><code>vignettes/renee.Rmd</code></a></small>
      <div class="d-none name"><code>renee.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CCBR/MOSuite" class="external-link">MOSuite</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="section level2">
<h2 id="renee-dataset">RENEE dataset<a class="anchor" aria-label="anchor" href="#renee-dataset"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># replace these lines with the actual paths to your files</span></span>
<span><span class="va">gene_counts_tsv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>  <span class="st">"RSEM.genes.expected_count.all_samples.txt.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"MOSuite"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">metadata_tsv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample_metadata.tsv.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"MOSuite"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create multi-omic object</span></span>
<span><span class="va">moo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_multiOmicDataSet_from_files.html">create_multiOmicDataSet_from_files</a></span><span class="op">(</span></span>
<span>  sample_meta_filepath <span class="op">=</span> <span class="va">metadata_tsv</span>,</span>
<span>  feature_counts_filepath <span class="op">=</span> <span class="va">gene_counts_tsv</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">58929</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">6</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (2): gene_id, GeneName</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (4): KO_S3, KO_S4, WT_S1, WT_S2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (2): sample_id, condition</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">moo</span><span class="op">@</span><span class="va">counts</span><span class="op">$</span><span class="va">raw</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   gene_id            KO_S3 KO_S4 WT_S1 WT_S2</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ENSG00000121410.11     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ENSG00000268895.5      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ENSG00000148584.15     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ENSG00000175899.14     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ENSG00000245105.3      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ENSG00000166535.20     0     0     0     0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">moo</span><span class="op">@</span><span class="va">sample_meta</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 2</span></span></span>
<span><span class="co">#&gt;   sample_id condition</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> KO_S3     knockout </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> KO_S4     knockout </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> WT_S1     wildtype </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> WT_S2     wildtype</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">moo</span><span class="op">@</span><span class="va">annotation</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   gene_id            GeneName</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ENSG00000121410.11 A1BG    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ENSG00000268895.5  A1BG-AS1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ENSG00000148584.15 A1CF    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ENSG00000175899.14 A2M     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ENSG00000245105.3  A2M-AS1 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ENSG00000166535.20 A2ML1</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/clean_raw_counts.html">clean_raw_counts</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_counts.html">filter_counts</a></span><span class="op">(</span></span>
<span>    group_colname <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>    label_colname <span class="op">=</span> <span class="st">"sample_id"</span>,</span>
<span>    minimum_count_value_to_be_considered_nonzero <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    minimum_number_of_samples_with_nonzero_counts_in_total <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    minimum_number_of_samples_with_nonzero_counts_in_a_group <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span></span>
<span>    group_colname <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>    label_colname <span class="op">=</span> <span class="st">"sample_id"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/diff_counts.html">diff_counts</a></span><span class="op">(</span></span>
<span>    covariates_colnames <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>    contrast_colname <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"knockout-wildtype"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_diff.html">filter_diff</a></span><span class="op">(</span></span>
<span>    significance_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    significance_column <span class="op">=</span> <span class="st">"adjpval"</span>,</span>
<span>    change_column <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>    change_cutoff <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; * cleaning raw counts</span></span>
<span><span class="co">#&gt; Not able to identify multiple id's in gene_id</span></span>
<span><span class="co">#&gt; Columns that can be used to aggregate gene information gene_id</span></span>
<span><span class="co">#&gt; Aggregating the counts for the same ID in different chromosome locations.</span></span>
<span><span class="co">#&gt; Column used to Aggregate duplicate IDs: gene_id</span></span>
<span><span class="co">#&gt; Number of rows before Collapse: 58929</span></span>
<span><span class="co">#&gt; no duplicated IDs in gene_id</span></span>
<span><span class="co">#&gt; * filtering clean counts</span></span>
<span><span class="co">#&gt; Number of features after filtering: 291</span></span>
<span><span class="co">#&gt; * normalizing filt counts</span></span>
<span><span class="co">#&gt; Total number of features included: 291</span></span>
<span><span class="co">#&gt; Sample columns: KO_S3, Sample columns: KO_S4, Sample columns: WT_S1, Sample columns: WT_S2</span></span>
<span><span class="co">#&gt; * differential counts</span></span>
<span><span class="co">#&gt; Setting first column of `counts` as gene annotation.</span></span>
<span><span class="co">#&gt; Total number of genes included: 291</span></span>
<span><span class="co">#&gt; * filtering differential features</span></span>
<span><span class="co">#&gt; Total number of genes selected with adjpval &lt; 0.05 and | logFC | ≥ 1 is sum(selgenes)</span></span>
<span></span>
<span><span class="va">moo</span><span class="op">@</span><span class="va">counts</span><span class="op">$</span><span class="va">norm</span><span class="op">$</span><span class="va">voom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;              gene_id     KO_S3     KO_S4     WT_S1     WT_S2</span></span>
<span><span class="co">#&gt; 1  ENSG00000215458.8 11.075196 12.348091  8.816153 10.004874</span></span>
<span><span class="co">#&gt; 2 ENSG00000160179.18  9.608634 12.770317 12.348091 12.236996</span></span>
<span><span class="co">#&gt; 3  ENSG00000258017.1  9.608634  8.816153  8.816153  8.816153</span></span>
<span><span class="co">#&gt; 4  ENSG00000282393.1  8.816153  9.608634  8.816153  8.816153</span></span>
<span><span class="co">#&gt; 5  ENSG00000286104.1  9.608634  8.816153  8.816153  8.816153</span></span>
<span><span class="co">#&gt; 6  ENSG00000274422.1  8.816153  9.608634  8.816153  8.816153</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-multiomicdataset-object-structure">The multiOmicDataSet object structure<a class="anchor" aria-label="anchor" href="#the-multiomicdataset-object-structure"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">moo</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt;MOSuite::multiOmicDataSet&gt;</span></span>
<span><span class="co">#&gt;  @ sample_meta: spc_tbl_ [4 × 2] (S3: spec_tbl_df/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ sample_id: chr [1:4] "KO_S3" "KO_S4" "WT_S1" "WT_S2"</span></span>
<span><span class="co">#&gt;  $ condition: chr [1:4] "knockout" "knockout" "wildtype" "wildtype"</span></span>
<span><span class="co">#&gt;  - attr(*, "spec")=</span></span>
<span><span class="co">#&gt;   .. cols(</span></span>
<span><span class="co">#&gt;   ..   sample_id = <span style="color: #BB0000;">col_character()</span>,</span></span>
<span><span class="co">#&gt;   ..   condition = <span style="color: #BB0000;">col_character()</span></span></span>
<span><span class="co">#&gt;   .. )</span></span>
<span><span class="co">#&gt;  - attr(*, "problems")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;  @ annotation : tibble [58,929 × 2] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ gene_id : chr [1:58929] "ENSG00000121410.11" "ENSG00000268895.5" "ENSG00000148584.15" "ENSG00000175899.14" ...</span></span>
<span><span class="co">#&gt;  $ GeneName: chr [1:58929] "A1BG" "A1BG-AS1" "A1CF" "A2M" ...</span></span>
<span><span class="co">#&gt;  @ counts     :List of 4</span></span>
<span><span class="co">#&gt;  .. $ raw  : tibble [58,929 × 5] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  ..  ..$ gene_id: chr [1:58929] "ENSG00000121410.11" "ENSG00000268895.5" "ENSG00000148584.15" "ENSG00000175899.14" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S3  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S4  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S1  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S2  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  .. $ clean:'data.frame':    58929 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ gene_id: chr [1:58929] "ENSG00000121410.11" "ENSG00000268895.5" "ENSG00000148584.15" "ENSG00000175899.14" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S3  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S4  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S1  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S2  : num [1:58929] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  .. $ filt :'data.frame':    291 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ gene_id: chr [1:291] "ENSG00000215458.8" "ENSG00000160179.18" "ENSG00000258017.1" "ENSG00000282393.1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S3  : num [1:291] 2 1 1 0 1 0 0 0 3 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ KO_S4  : num [1:291] 4 5 0 1 0 1 0 0 3 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S1  : num [1:291] 0 6 0 0 0 0 46 33 9 0 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ WT_S2  : num [1:291] 1 7 0 0 0 0 80 31 11 1 ...</span></span>
<span><span class="co">#&gt;  .. $ norm :List of 1</span></span>
<span><span class="co">#&gt;  ..  ..$ voom:'data.frame':  291 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ gene_id: chr [1:291] "ENSG00000215458.8" "ENSG00000160179.18" "ENSG00000258017.1" "ENSG00000282393.1" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ KO_S3  : num [1:291] 11.08 9.61 9.61 8.82 9.61 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ KO_S4  : num [1:291] 12.35 12.77 8.82 9.61 8.82 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ WT_S1  : num [1:291] 8.82 12.35 8.82 8.82 8.82 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ WT_S2  : num [1:291] 10 12.24 8.82 8.82 8.82 ...</span></span>
<span><span class="co">#&gt;  @ analyses   :List of 3</span></span>
<span><span class="co">#&gt;  .. $ colors   :List of 2</span></span>
<span><span class="co">#&gt;  ..  ..$ sample_id: Named chr [1:4] "#000000" "#E69F00" "#56B4E9" "#009E73"</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:4] "KO_S3" "KO_S4" "WT_S1" "WT_S2"</span></span>
<span><span class="co">#&gt;  ..  ..$ condition: Named chr [1:2] "#000000" "#E69F00"</span></span>
<span><span class="co">#&gt;  ..  .. ..- attr(*, "names")= chr [1:2] "knockout" "wildtype"</span></span>
<span><span class="co">#&gt;  .. $ diff     :List of 1</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype:'data.frame': 291 obs. of  6 variables:</span></span>
<span><span class="co">#&gt;  ..  .. ..$ gene_id: chr [1:291] "ENSG00000215458.8" "ENSG00000160179.18" "ENSG00000258017.1" "ENSG00000282393.1" ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ FC     : num [1:291] 4.69 -2.13 1.32 1.32 1.32 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ logFC  : num [1:291] 2.23 -1.09 0.396 0.396 0.396 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ tstat  : num [1:291] 2.956 -0.957 0.557 0.557 0.557 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ pval   : num [1:291] 0.0648 0.4135 0.6188 0.6188 0.6188 ...</span></span>
<span><span class="co">#&gt;  ..  .. ..$ adjpval: num [1:291] 0.185 0.658 0.69 0.69 0.69 ...</span></span>
<span><span class="co">#&gt;  .. $ diff_filt:'data.frame':    54 obs. of  6 variables:</span></span>
<span><span class="co">#&gt;  ..  ..$ gene_id                  : chr [1:54] "ENSG00000154734.15" "ENSG00000154736.6" "ENSG00000232855.6" "ENSG00000231324.1" ...</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype_FC     : num [1:54] -54.2 -35.5 -7.17 -2.77 6.7 -6.39 -27.8 -5.62 6.25 6.7 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype_logFC  : num [1:54] -5.76 -5.15 -2.84 -1.47 2.74 -2.68 -4.8 -2.49 2.64 2.74 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype_tstat  : num [1:54] -103 -31.1 -14.8 -6.67 8.72 -13.4 -14.4 -11.1 6.94 8.72 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype_pval   : num [1:54] 4.16e-06 1.19e-04 9.54e-04 8.40e-03 4.07e-03 1.26e-03 1.02e-03 2.09e-03 7.57e-03 4.07e-03 ...</span></span>
<span><span class="co">#&gt;  ..  ..$ knockout-wildtype_adjpval: num [1:54] 0.000695 0.00493 0.0139 0.0479 0.0296 0.0146 0.014 0.0194 0.0468 0.0296 ...</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer>
</body>
</html>
