<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Differential expression analysis — diff_counts • MOSuite</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential expression analysis — diff_counts"><meta name="description" content="Differential expression analysis"><meta property="og:description" content="Differential expression analysis"></head><body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Unreleased version">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic" id="container">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Differential expression analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/R/deg.R" class="external-link"><code>R/deg.R</code></a></small>
      <div class="d-none name"><code>diff_counts.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Differential expression analysis</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">diff_counts</span><span class="op">(</span></span>
<span>  <span class="va">moo</span>,</span>
<span>  count_type <span class="op">=</span> <span class="st">"filt"</span>,</span>
<span>  sub_count_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sample_id_colname <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  feature_id_colname <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  samples_to_include <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  covariates_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span>
<span>  contrast_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span>
<span>  contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A"</span>, <span class="st">"C-A"</span>, <span class="st">"B-C"</span><span class="op">)</span>,</span>
<span>  input_in_log_counts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  return_mean_and_sd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  return_normalized_counts <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  voom_normalization_method <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>  print_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"print_plots"</span><span class="op">)</span>,</span>
<span>  save_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"save_plots"</span><span class="op">)</span>,</span>
<span>  plots_subdir <span class="op">=</span> <span class="st">"diff"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-moo">moo<a class="anchor" aria-label="anchor" href="#arg-moo"></a></dt>
<dd><p>multiOmicDataSet object (see <code><a href="create_multiOmicDataSet_from_dataframes.html">create_multiOmicDataSet_from_dataframes()</a></code>)</p></dd>


<dt id="arg-count-type">count_type<a class="anchor" aria-label="anchor" href="#arg-count-type"></a></dt>
<dd><p>the type of counts to use – must be a name in the counts slot (<code>moo@counts</code>)</p></dd>


<dt id="arg-sub-count-type">sub_count_type<a class="anchor" aria-label="anchor" href="#arg-sub-count-type"></a></dt>
<dd><p>if <code>count_type</code> is a list, specify the sub count type within the list. (Default: <code>NULL</code>)</p></dd>


<dt id="arg-sample-id-colname">sample_id_colname<a class="anchor" aria-label="anchor" href="#arg-sample-id-colname"></a></dt>
<dd><p>The column from the sample metadata containing the sample names. The names in this column must exactly match the names used as the sample column names of your input Counts Matrix. (Default: <code>NULL</code> - first column in the sample metadata will be used.)</p></dd>


<dt id="arg-feature-id-colname">feature_id_colname<a class="anchor" aria-label="anchor" href="#arg-feature-id-colname"></a></dt>
<dd><p>The column from the counts dataa containing the Feature IDs (Usually Gene or Protein ID). This is usually the first column of your input Counts Matrix. Only columns of Text type from your input Counts Matrix will be available to select for this parameter. (Default: <code>NULL</code> - first column in the counts matrix will be used.)</p></dd>


<dt id="arg-samples-to-include">samples_to_include<a class="anchor" aria-label="anchor" href="#arg-samples-to-include"></a></dt>
<dd><p>Which samples would you like to include? Usually, you will choose all sample columns, or you could choose to remove certain samples. Samples excluded here will be removed in this step and from further analysis downstream of this step. (Default: <code>NULL</code> - all sample IDs in <code>moo@sample_meta</code> will be used.)</p></dd>


<dt id="arg-covariates-colnames">covariates_colnames<a class="anchor" aria-label="anchor" href="#arg-covariates-colnames"></a></dt>
<dd><p>Columns to be used as covariates in linear modeling. Must include column from "Contrast Variable". Most commonly your covariate will be group and batch (if you have different batches in your data).</p></dd>


<dt id="arg-contrast-colname">contrast_colname<a class="anchor" aria-label="anchor" href="#arg-contrast-colname"></a></dt>
<dd><p>The column in the metadata that contains the group variables you wish to find differential expression between. Up to 2 columns (2-factor analysis) can be used.</p></dd>


<dt id="arg-contrasts">contrasts<a class="anchor" aria-label="anchor" href="#arg-contrasts"></a></dt>
<dd><p>Specify each contrast in the format group1-group2, e.g. treated-control</p></dd>


<dt id="arg-input-in-log-counts">input_in_log_counts<a class="anchor" aria-label="anchor" href="#arg-input-in-log-counts"></a></dt>
<dd><p>set this to <code>TRUE</code> if counts are already log2-transformed</p></dd>


<dt id="arg-return-mean-and-sd">return_mean_and_sd<a class="anchor" aria-label="anchor" href="#arg-return-mean-and-sd"></a></dt>
<dd><p>if TRUE, return Mean and Standard Deviation of groups in addition to DEG estimates for contrast(s)</p></dd>


<dt id="arg-return-normalized-counts">return_normalized_counts<a class="anchor" aria-label="anchor" href="#arg-return-normalized-counts"></a></dt>
<dd><p>if TRUE, return normalized counts for samples included in the limma model</p></dd>


<dt id="arg-voom-normalization-method">voom_normalization_method<a class="anchor" aria-label="anchor" href="#arg-voom-normalization-method"></a></dt>
<dd><p>Normalization method to be applied to the logCPM values when using <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom</a></code></p></dd>


<dt id="arg-print-plots">print_plots<a class="anchor" aria-label="anchor" href="#arg-print-plots"></a></dt>
<dd><p>Whether to print plots during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_print_plots' or environment variable 'MOO_PRINT_PLOTS')</p></dd>


<dt id="arg-save-plots">save_plots<a class="anchor" aria-label="anchor" href="#arg-save-plots"></a></dt>
<dd><p>Whether to save plots to files during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_save_plots' or environment variable 'MOO_SAVE_PLOTS')</p></dd>


<dt id="arg-plots-subdir">plots_subdir<a class="anchor" aria-label="anchor" href="#arg-plots-subdir"></a></dt>
<dd><p>subdirectory in where plots will be saved if <code>save_plots</code> is <code>TRUE</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>multiOmicDataSet</code> with <code>diff</code> added to the <code>analyses</code> slot (i.e. <code>moo@analyses$diff</code>)</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">moo</span> <span class="op">&lt;-</span> <span class="fu"><a href="multiOmicDataSet.html">multiOmicDataSet</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_sample_metadata</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  anno_dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  counts_lst <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"raw"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_raw_counts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"clean"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_clean_raw_counts</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="st">"filt"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_filtered_counts</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu">diff_counts</span><span class="op">(</span></span></span>
<span class="r-in"><span>    count_type <span class="op">=</span> <span class="st">"filt"</span>,</span></span>
<span class="r-in"><span>    sub_count_type <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>    sample_id_colname <span class="op">=</span> <span class="st">"Sample"</span>,</span></span>
<span class="r-in"><span>    feature_id_colname <span class="op">=</span> <span class="st">"Gene"</span>,</span></span>
<span class="r-in"><span>    covariates_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    contrast_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A"</span>, <span class="st">"C-A"</span>, <span class="st">"B-C"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    voom_normalization_method <span class="op">=</span> <span class="st">"quantile"</span>,</span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> * differential counts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting first column of `counts` as gene annotation.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total number of genes included: 7943</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">moo</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            Gene    B-A_FC    C-A_FC    B-C_FC   B-A_logFC   C-A_logFC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0610007P14Rik -1.026892  1.106240 -1.135989 -0.03828439  0.14566407</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0610009B22Rik -1.009168 -1.267764  1.256247 -0.01316628 -0.34228628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 0610010F05Rik  1.122161 -6.063562  6.804293  0.16627966 -2.60016562</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 0610011F06Rik  1.179418  1.357548 -1.151032  0.23807481  0.44100333</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 0610012G03Rik -1.180719  1.051413 -1.241423 -0.23966515  0.07232912</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 0610037L13Rik -1.422360 -1.429566  1.005067 -0.50828657 -0.51557755</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      B-C_logFC   B-A_tstat  C-A_tstat   B-C_tstat  B-A_pval    C-A_pval</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 -0.183948458 -0.11239064  0.4619930 -0.58386927 0.9124155 0.652543422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  0.329120003 -0.01744031 -0.4779059  0.42923631 0.9863787 0.641505931</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  2.766445272  0.29807039 -3.0878622  3.11529979 0.7708668 0.009637891</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 -0.202928519  0.43922204  0.8994597 -0.42166164 0.6684882 0.386513298</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 -0.311994268 -0.37124882  0.1260545 -0.52269647 0.7170755 0.901825097</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  0.007290977 -0.73372174 -0.7907409  0.01033402 0.4775382 0.444800236</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      B-C_pval B-A_adjpval C-A_adjpval B-C_adjpval       A1       A2       A3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.570373641   0.9822240  0.82272260   0.8159825 6.532994 6.192871 5.954869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.675534876   0.9973532  0.81775923   0.8721628 4.484983 5.448875 5.286875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 0.009163897   0.9577678  0.07958189   0.1269944 4.883688 5.668494 6.537590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 0.680901716   0.9286117  0.62868219   0.8742464 5.199684 5.374085 5.112952</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 0.610922936   0.9499111  0.95865237   0.8370657 5.368118 5.445918 5.456511</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 0.991928593   0.8709056  0.67812827   0.9966969 5.327987 5.388747 5.233520</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         B1       B2       B3       C1       C2       C3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 6.375896 6.275880 6.119449 6.419913 6.172204 6.497050</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 3.445612 4.451347 5.473886 3.500359 4.709254 5.471951</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 6.216408 5.893089 5.498884 3.845207 2.685177 2.805426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 5.155558 5.163359 5.650929 5.441965 6.043492 5.490958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 4.567138 5.274928 5.625039 5.787457 6.214163 4.682896</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 5.450169 3.656585 4.929386 4.274944 4.744405 5.173531</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer></body></html>

