<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Enhanced Volcano Plot — plot_volcano_enhanced • MOSuite</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Enhanced Volcano Plot — plot_volcano_enhanced"><meta name="description" content="Uses Bioconductor's Enhanced Volcano Plot."><meta property="og:description" content="Uses Bioconductor's Enhanced Volcano Plot."></head><body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Unreleased version">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic" id="container">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Enhanced Volcano Plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/R/plot_volcano_enhanced.R" class="external-link"><code>R/plot_volcano_enhanced.R</code></a></small>
      <div class="d-none name"><code>plot_volcano_enhanced.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Uses <a href="https://bioconductor.org/packages/release/bioc/html/EnhancedVolcano.html" class="external-link">Bioconductor's Enhanced Volcano Plot</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_volcano_enhanced</span><span class="op">(</span></span>
<span>  <span class="va">moo_diff</span>,</span>
<span>  feature_id_colname <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  signif_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A_adjpval"</span>, <span class="st">"B-C_adjpval"</span><span class="op">)</span>,</span>
<span>  signif_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  change_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A_logFC"</span>, <span class="st">"B-C_logFC"</span><span class="op">)</span>,</span>
<span>  change_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  value_to_sort_the_output_dataset <span class="op">=</span> <span class="st">"p-value"</span>,</span>
<span>  num_features_to_label <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  use_only_addition_labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  additional_labels <span class="op">=</span> <span class="st">""</span>,</span>
<span>  is_red <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  lab_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  change_sig_name <span class="op">=</span> <span class="st">"p-value"</span>,</span>
<span>  change_lfc_name <span class="op">=</span> <span class="st">"log2FC"</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Volcano Plots"</span>,</span>
<span>  use_custom_lab <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ylim <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  custom_xlim <span class="op">=</span> <span class="st">""</span>,</span>
<span>  xlim_additional <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  ylim_additional <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  axis_lab_size <span class="op">=</span> <span class="fl">24</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  image_width <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  image_height <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  interactive_plots <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  print_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"print_plots"</span><span class="op">)</span>,</span>
<span>  save_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"save_plots"</span><span class="op">)</span>,</span>
<span>  plots_subdir <span class="op">=</span> <span class="st">"diff"</span>,</span>
<span>  plot_filename <span class="op">=</span> <span class="st">"volcano_enhanced.png"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-moo-diff">moo_diff<a class="anchor" aria-label="anchor" href="#arg-moo-diff"></a></dt>
<dd><p>Differential expression analysis result from one or more contrasts</p></dd>


<dt id="arg-feature-id-colname">feature_id_colname<a class="anchor" aria-label="anchor" href="#arg-feature-id-colname"></a></dt>
<dd><p>The column from the counts data containing the Feature IDs (Usually Gene or Protein ID). This is usually the first column of your input Counts Matrix. Only columns of Text type from your input Counts Matrix will be available to select for this parameter. (Default: <code>NULL</code> - first column in the counts matrix will be used.)</p></dd>


<dt id="arg-signif-colname">signif_colname<a class="anchor" aria-label="anchor" href="#arg-signif-colname"></a></dt>
<dd><p>column name of significance values (e.g., adjusted p-values or FDR). This column will be used to determine which points are considered significant in the volcano plot.</p></dd>


<dt id="arg-signif-threshold">signif_threshold<a class="anchor" aria-label="anchor" href="#arg-signif-threshold"></a></dt>
<dd><p>Numeric value specifying the significance cutoff for p-values (i.e. filters on <code>signif_colname</code>)</p></dd>


<dt id="arg-change-colname">change_colname<a class="anchor" aria-label="anchor" href="#arg-change-colname"></a></dt>
<dd><p>column name of fold change values.</p></dd>


<dt id="arg-change-threshold">change_threshold<a class="anchor" aria-label="anchor" href="#arg-change-threshold"></a></dt>
<dd><p>Numeric value specifying the fold change cutoff for significance (i.e. filters on <code>change_colname</code>)</p></dd>


<dt id="arg-value-to-sort-the-output-dataset">value_to_sort_the_output_dataset<a class="anchor" aria-label="anchor" href="#arg-value-to-sort-the-output-dataset"></a></dt>
<dd><p>How to sort the output dataset. Options are "fold-change" or "p-value".</p></dd>


<dt id="arg-num-features-to-label">num_features_to_label<a class="anchor" aria-label="anchor" href="#arg-num-features-to-label"></a></dt>
<dd><p>Number of top features/genes to label in the volcano plot. Default is 30.</p></dd>


<dt id="arg-use-only-addition-labels">use_only_addition_labels<a class="anchor" aria-label="anchor" href="#arg-use-only-addition-labels"></a></dt>
<dd><p>If <code>TRUE</code>, only the additional labels specified in <code>additional_labels</code> will be used for labeling in the volcano plot, ignoring the top features.</p></dd>


<dt id="arg-additional-labels">additional_labels<a class="anchor" aria-label="anchor" href="#arg-additional-labels"></a></dt>
<dd><p>comma-separated string of feature names or IDs to include in the volcano plot.</p></dd>


<dt id="arg-is-red">is_red<a class="anchor" aria-label="anchor" href="#arg-is-red"></a></dt>
<dd><p>Logical. If TRUE, highlights points in red.</p></dd>


<dt id="arg-lab-size">lab_size<a class="anchor" aria-label="anchor" href="#arg-lab-size"></a></dt>
<dd><p>Size of the labels in the volcano plot.</p></dd>


<dt id="arg-change-sig-name">change_sig_name<a class="anchor" aria-label="anchor" href="#arg-change-sig-name"></a></dt>
<dd><p>Name for the significance column in the plot. Default is "p-value".</p></dd>


<dt id="arg-change-lfc-name">change_lfc_name<a class="anchor" aria-label="anchor" href="#arg-change-lfc-name"></a></dt>
<dd><p>Name for the fold change column in the plot. Default is "log2FC".</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>Title of the plot. Default is "Volcano Plots".</p></dd>


<dt id="arg-use-custom-lab">use_custom_lab<a class="anchor" aria-label="anchor" href="#arg-use-custom-lab"></a></dt>
<dd><p>If TRUE, uses custom labels for the plot.</p></dd>


<dt id="arg-ylim">ylim<a class="anchor" aria-label="anchor" href="#arg-ylim"></a></dt>
<dd><p>Y-axis limits for the plot.</p></dd>


<dt id="arg-custom-xlim">custom_xlim<a class="anchor" aria-label="anchor" href="#arg-custom-xlim"></a></dt>
<dd><p>Custom X-axis limits for the plot.</p></dd>


<dt id="arg-xlim-additional">xlim_additional<a class="anchor" aria-label="anchor" href="#arg-xlim-additional"></a></dt>
<dd><p>Additional space to add to the X-axis limits.</p></dd>


<dt id="arg-ylim-additional">ylim_additional<a class="anchor" aria-label="anchor" href="#arg-ylim-additional"></a></dt>
<dd><p>Additional space to add to the Y-axis limits.</p></dd>


<dt id="arg-axis-lab-size">axis_lab_size<a class="anchor" aria-label="anchor" href="#arg-axis-lab-size"></a></dt>
<dd><p>Size of the axis labels.</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Size of the points in the plot.</p></dd>


<dt id="arg-image-width">image_width<a class="anchor" aria-label="anchor" href="#arg-image-width"></a></dt>
<dd><p>output image width in pixels - only used if save_plots is TRUE</p></dd>


<dt id="arg-image-height">image_height<a class="anchor" aria-label="anchor" href="#arg-image-height"></a></dt>
<dd><p>output image height in pixels - only used if save_plots is TRUE</p></dd>


<dt id="arg-dpi">dpi<a class="anchor" aria-label="anchor" href="#arg-dpi"></a></dt>
<dd><p>dots-per-inch of the output image (see <code>ggsave()</code>) - only used if save_plots is TRUE</p></dd>


<dt id="arg-interactive-plots">interactive_plots<a class="anchor" aria-label="anchor" href="#arg-interactive-plots"></a></dt>
<dd><p>set to TRUE to make PCA and Histogram plots interactive with <code>plotly</code>, allowing you to hover your mouse over a point or line to view sample information. The similarity heat map will not display if this toggle is set to TRUE. Default is FALSE.</p></dd>


<dt id="arg-print-plots">print_plots<a class="anchor" aria-label="anchor" href="#arg-print-plots"></a></dt>
<dd><p>Whether to print plots during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_print_plots' or environment variable 'MOO_PRINT_PLOTS')</p></dd>


<dt id="arg-save-plots">save_plots<a class="anchor" aria-label="anchor" href="#arg-save-plots"></a></dt>
<dd><p>Whether to save plots to files during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_save_plots' or environment variable 'MOO_SAVE_PLOTS')</p></dd>


<dt id="arg-plots-subdir">plots_subdir<a class="anchor" aria-label="anchor" href="#arg-plots-subdir"></a></dt>
<dd><p>subdirectory in where plots will be saved if <code>save_plots</code> is <code>TRUE</code></p></dd>


<dt id="arg-plot-filename">plot_filename<a class="anchor" aria-label="anchor" href="#arg-plot-filename"></a></dt>
<dd><p>plot output filename - only used if save_plots is TRUE</p></dd>

</dl></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer></body></html>

