<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Volcano Plot - Summary [CC Produces one volcano plot for each tested contrast in the input DEG table. — plot_volcano_summary • MOSuite</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Volcano Plot - Summary [CC Produces one volcano plot for each tested contrast in the input DEG table. — plot_volcano_summary"><meta name="description" content="It can be sorted by either fold change, t-statistic, or p-value. The returned
dataset includes one row for each significant gene in each contrast, and
contains columns from the DEG analysis of that contrast as well as columns
useful to the Venn diagram template downstream."><meta property="og:description" content="It can be sorted by either fold change, t-statistic, or p-value. The returned
dataset includes one row for each significant gene in each contrast, and
contains columns from the DEG analysis of that contrast as well as columns
useful to the Venn diagram template downstream."></head><body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Unreleased version">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic" id="container">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Volcano Plot - Summary [CC Produces one volcano plot for each tested contrast in the input DEG table.</h1>
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/R/plot_volcano_summary.R" class="external-link"><code>R/plot_volcano_summary.R</code></a></small>
      <div class="d-none name"><code>plot_volcano_summary.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>It can be sorted by either fold change, t-statistic, or p-value. The returned
dataset includes one row for each significant gene in each contrast, and
contains columns from the DEG analysis of that contrast as well as columns
useful to the Venn diagram template downstream.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_volcano_summary</span><span class="op">(</span></span>
<span>  <span class="va">moo_diff</span>,</span>
<span>  feature_id_colname <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  signif_colname <span class="op">=</span> <span class="st">"pval"</span>,</span>
<span>  signif_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  change_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  value_to_sort_the_output_dataset <span class="op">=</span> <span class="st">"t-statistic"</span>,</span>
<span>  num_features_to_label <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  add_features <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  label_features <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  custom_gene_list <span class="op">=</span> <span class="st">""</span>,</span>
<span>  default_label_color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  custom_label_color <span class="op">=</span> <span class="st">"green3"</span>,</span>
<span>  label_x_adj <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  label_y_adj <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  line_thickness <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  label_font_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  label_font_type <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  displace_feature_labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  custom_gene_list_special_label_displacement <span class="op">=</span> <span class="st">""</span>,</span>
<span>  special_label_displacement_x_axis <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  special_label_displacement_y_axis <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  color_of_signif_threshold_line <span class="op">=</span> <span class="st">"blue"</span>,</span>
<span>  color_of_non_significant_features <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  color_of_logfold_change_threshold_line <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  color_of_features_meeting_only_signif_threshold <span class="op">=</span> <span class="st">"lightgoldenrod2"</span>,</span>
<span>  color_for_features_meeting_pvalue_and_foldchange_thresholds <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  flip_vplot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_default_x_axis_limit <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  x_axis_limit <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  use_default_y_axis_limit <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  y_axis_limit <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  add_deg_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FC"</span>, <span class="st">"logFC"</span>, <span class="st">"tstat"</span>, <span class="st">"pval"</span>, <span class="st">"adjpval"</span><span class="op">)</span>,</span>
<span>  graphics_device <span class="op">=</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span>,</span>
<span>  image_width <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  image_height <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  use_default_grid_layout <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  number_of_rows_in_grid_layout <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  aspect_ratio <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  plot_filename <span class="op">=</span> <span class="st">"volcano.png"</span>,</span>
<span>  print_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"print_plots"</span><span class="op">)</span>,</span>
<span>  save_plots <span class="op">=</span> <span class="fu">options</span><span class="fu">::</span><span class="fu"><a href="https://dgkf.github.io/options/reference/opt.html" class="external-link">opt</a></span><span class="op">(</span><span class="st">"save_plots"</span><span class="op">)</span>,</span>
<span>  plots_subdir <span class="op">=</span> <span class="st">"figures"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-moo-diff">moo_diff<a class="anchor" aria-label="anchor" href="#arg-moo-diff"></a></dt>
<dd><p>Differential expression analysis result from one or more contrasts</p></dd>


<dt id="arg-feature-id-colname">feature_id_colname<a class="anchor" aria-label="anchor" href="#arg-feature-id-colname"></a></dt>
<dd><p>The column from the counts data containing the Feature IDs (Usually Gene or Protein ID). This is usually the first column of your input Counts Matrix. Only columns of Text type from your input Counts Matrix will be available to select for this parameter. (Default: <code>NULL</code> - first column in the counts matrix will be used.)</p></dd>


<dt id="arg-signif-colname">signif_colname<a class="anchor" aria-label="anchor" href="#arg-signif-colname"></a></dt>
<dd><p>column name of significance values (e.g., adjusted p-values or FDR). This column will be used to determine which points are considered significant in the volcano plot.</p></dd>


<dt id="arg-signif-threshold">signif_threshold<a class="anchor" aria-label="anchor" href="#arg-signif-threshold"></a></dt>
<dd><p>Numeric value specifying the significance cutoff for p-values (i.e. filters on <code>signif_colname</code>)</p></dd>


<dt id="arg-change-threshold">change_threshold<a class="anchor" aria-label="anchor" href="#arg-change-threshold"></a></dt>
<dd><p>Numeric value specifying the fold change cutoff for significance (i.e. filters on <code>change_colname</code>)</p></dd>


<dt id="arg-value-to-sort-the-output-dataset">value_to_sort_the_output_dataset<a class="anchor" aria-label="anchor" href="#arg-value-to-sort-the-output-dataset"></a></dt>
<dd><p>How to sort the output dataset. Options are "fold-change" or "p-value".</p></dd>


<dt id="arg-num-features-to-label">num_features_to_label<a class="anchor" aria-label="anchor" href="#arg-num-features-to-label"></a></dt>
<dd><p>Number of top features/genes to label in the volcano plot. Default is 30.</p></dd>


<dt id="arg-add-features">add_features<a class="anchor" aria-label="anchor" href="#arg-add-features"></a></dt>
<dd><p>Add custom_gene_list To Labels. Set TRUE when you want to label a specific set of features (features) in the "custom_gene_list" parameter" IN ADDITION to the number of features you set in the "Number of Features to Label" parameter.</p></dd>


<dt id="arg-label-features">label_features<a class="anchor" aria-label="anchor" href="#arg-label-features"></a></dt>
<dd><p>Select TRUE when you want to label ONLY a specific list of features(features) given in the "custom_gene_list" parameter.</p></dd>


<dt id="arg-custom-gene-list">custom_gene_list<a class="anchor" aria-label="anchor" href="#arg-custom-gene-list"></a></dt>
<dd><p>Provide a list of features (comma separated) to be labeled on the volcano plot. You must toggle one of the following ON to see these labels: "Add features" or "Label Only My Feature List".</p></dd>


<dt id="arg-default-label-color">default_label_color<a class="anchor" aria-label="anchor" href="#arg-default-label-color"></a></dt>
<dd><p>Set the color for the text used to add feature (gene) name labels to points.</p></dd>


<dt id="arg-custom-label-color">custom_label_color<a class="anchor" aria-label="anchor" href="#arg-custom-label-color"></a></dt>
<dd><p>Set the color for the specific list of features (features) provided in the "Feature List" parameter.</p></dd>


<dt id="arg-label-x-adj">label_x_adj<a class="anchor" aria-label="anchor" href="#arg-label-x-adj"></a></dt>
<dd><p>adjust position of the labels on the x-axis. Default: 0.2</p></dd>


<dt id="arg-label-y-adj">label_y_adj<a class="anchor" aria-label="anchor" href="#arg-label-y-adj"></a></dt>
<dd><p>adjust position of the labels on the y-axis. Default: 0.2</p></dd>


<dt id="arg-line-thickness">line_thickness<a class="anchor" aria-label="anchor" href="#arg-line-thickness"></a></dt>
<dd><p>Set the thickness of the lines in the plot. Default: 0.5</p></dd>


<dt id="arg-label-font-size">label_font_size<a class="anchor" aria-label="anchor" href="#arg-label-font-size"></a></dt>
<dd><p>Set the font size of the labels. Default: 4</p></dd>


<dt id="arg-label-font-type">label_font_type<a class="anchor" aria-label="anchor" href="#arg-label-font-type"></a></dt>
<dd><p>Set the font type of the labels. Default: 1</p></dd>


<dt id="arg-displace-feature-labels">displace_feature_labels<a class="anchor" aria-label="anchor" href="#arg-displace-feature-labels"></a></dt>
<dd><p>Set to TRUE to displace gene labels. Default: FALSE. Set TRUE if you want to displace the feature (gene) label for a specific set of features. Make sure to use custom x- and y- limits and give sufficient space for displacement; otherwise other labels than the desired ones will appear displaced.</p></dd>


<dt id="arg-custom-gene-list-special-label-displacement">custom_gene_list_special_label_displacement<a class="anchor" aria-label="anchor" href="#arg-custom-gene-list-special-label-displacement"></a></dt>
<dd><p>Provide a list of features (comma separated) for which you want special displacement of the feature label.</p></dd>


<dt id="arg-special-label-displacement-x-axis">special_label_displacement_x_axis<a class="anchor" aria-label="anchor" href="#arg-special-label-displacement-x-axis"></a></dt>
<dd><p>Displacement of the feature label on the x-axis. Default: 2</p></dd>


<dt id="arg-special-label-displacement-y-axis">special_label_displacement_y_axis<a class="anchor" aria-label="anchor" href="#arg-special-label-displacement-y-axis"></a></dt>
<dd><p>Displacement of the feature label on the y-axis. Default: 2</p></dd>


<dt id="arg-color-of-signif-threshold-line">color_of_signif_threshold_line<a class="anchor" aria-label="anchor" href="#arg-color-of-signif-threshold-line"></a></dt>
<dd><p>Color of the significance threshold line. Default: "blue"</p></dd>


<dt id="arg-color-of-non-significant-features">color_of_non_significant_features<a class="anchor" aria-label="anchor" href="#arg-color-of-non-significant-features"></a></dt>
<dd><p>Color of the non-significant features. Default: "black"</p></dd>


<dt id="arg-color-of-logfold-change-threshold-line">color_of_logfold_change_threshold_line<a class="anchor" aria-label="anchor" href="#arg-color-of-logfold-change-threshold-line"></a></dt>
<dd><p>Color of the log fold change threshold line. Default: "red"</p></dd>


<dt id="arg-color-of-features-meeting-only-signif-threshold">color_of_features_meeting_only_signif_threshold<a class="anchor" aria-label="anchor" href="#arg-color-of-features-meeting-only-signif-threshold"></a></dt>
<dd><p>Color of the features that meet only the significance threshold. Default: "lightgoldenrod2"</p></dd>


<dt id="arg-color-for-features-meeting-pvalue-and-foldchange-thresholds">color_for_features_meeting_pvalue_and_foldchange_thresholds<a class="anchor" aria-label="anchor" href="#arg-color-for-features-meeting-pvalue-and-foldchange-thresholds"></a></dt>
<dd><p>Color of the features that meet both the p-value and fold change thresholds. Default: "red"</p></dd>


<dt id="arg-flip-vplot">flip_vplot<a class="anchor" aria-label="anchor" href="#arg-flip-vplot"></a></dt>
<dd><p>Set to TRUE to flip the fold change values so that the volcano plot looks like a comparison was B-A. Default: FALSE</p></dd>


<dt id="arg-use-default-x-axis-limit">use_default_x_axis_limit<a class="anchor" aria-label="anchor" href="#arg-use-default-x-axis-limit"></a></dt>
<dd><p>Set to TRUE to use the default x-axis limit. Default: TRUE</p></dd>


<dt id="arg-x-axis-limit">x_axis_limit<a class="anchor" aria-label="anchor" href="#arg-x-axis-limit"></a></dt>
<dd><p>Custom x-axis limit. Default: c(-5, 5)</p></dd>


<dt id="arg-use-default-y-axis-limit">use_default_y_axis_limit<a class="anchor" aria-label="anchor" href="#arg-use-default-y-axis-limit"></a></dt>
<dd><p>Set to TRUE to use the default y-axis limit. Default: TRUE</p></dd>


<dt id="arg-y-axis-limit">y_axis_limit<a class="anchor" aria-label="anchor" href="#arg-y-axis-limit"></a></dt>
<dd><p>Custom y-axis limit. Default: c(0, 10)</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Size of the points in the plot. Default: 1</p></dd>


<dt id="arg-add-deg-columns">add_deg_columns<a class="anchor" aria-label="anchor" href="#arg-add-deg-columns"></a></dt>
<dd><p>Add additional columns from the DEG analysis to the output dataset. Default: FALSE</p></dd>


<dt id="arg-graphics-device">graphics_device<a class="anchor" aria-label="anchor" href="#arg-graphics-device"></a></dt>
<dd><p>passed to <code>ggsave(device)</code>. Default: <code><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">grDevices::png</a></code></p></dd>


<dt id="arg-image-width">image_width<a class="anchor" aria-label="anchor" href="#arg-image-width"></a></dt>
<dd><p>output image width in pixels - only used if save_plots is TRUE</p></dd>


<dt id="arg-image-height">image_height<a class="anchor" aria-label="anchor" href="#arg-image-height"></a></dt>
<dd><p>output image height in pixels - only used if save_plots is TRUE</p></dd>


<dt id="arg-dpi">dpi<a class="anchor" aria-label="anchor" href="#arg-dpi"></a></dt>
<dd><p>dots-per-inch of the output image (see <code>ggsave()</code>) - only used if save_plots is TRUE</p></dd>


<dt id="arg-use-default-grid-layout">use_default_grid_layout<a class="anchor" aria-label="anchor" href="#arg-use-default-grid-layout"></a></dt>
<dd><p>Set to TRUE to use the default grid layout. Default: TRUE</p></dd>


<dt id="arg-number-of-rows-in-grid-layout">number_of_rows_in_grid_layout<a class="anchor" aria-label="anchor" href="#arg-number-of-rows-in-grid-layout"></a></dt>
<dd><p>Number of rows in the grid layout. Default: 1</p></dd>


<dt id="arg-aspect-ratio">aspect_ratio<a class="anchor" aria-label="anchor" href="#arg-aspect-ratio"></a></dt>
<dd><p>Aspect ratio of the output image. Default: 4/3</p></dd>


<dt id="arg-plot-filename">plot_filename<a class="anchor" aria-label="anchor" href="#arg-plot-filename"></a></dt>
<dd><p>Filename for the output plot. Default: "volcano_plot.png"</p></dd>


<dt id="arg-print-plots">print_plots<a class="anchor" aria-label="anchor" href="#arg-print-plots"></a></dt>
<dd><p>Whether to print plots during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_print_plots' or environment variable 'MOO_PRINT_PLOTS')</p></dd>


<dt id="arg-save-plots">save_plots<a class="anchor" aria-label="anchor" href="#arg-save-plots"></a></dt>
<dd><p>Whether to save plots to files during analysis (Defaults to <code>FALSE</code>, overwritable using option 'moo_save_plots' or environment variable 'MOO_SAVE_PLOTS')</p></dd>


<dt id="arg-plots-subdir">plots_subdir<a class="anchor" aria-label="anchor" href="#arg-plots-subdir"></a></dt>
<dd><p>subdirectory in where plots will be saved if <code>save_plots</code> is <code>TRUE</code></p></dd>

</dl></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer></body></html>

