<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Remove low-count genes — remove_low_count_genes • MOSuite</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Remove low-count genes — remove_low_count_genes"><meta name="description" content="TODO this function also transforms raw counts to CPM, but that should be a separate function before this step, before filter_counts function()"><meta property="og:description" content="TODO this function also transforms raw counts to CPM, but that should be a separate function before this step, before filter_counts function()"></head><body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Unreleased version">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic" id="container">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Remove low-count genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/main/R/filter.R" class="external-link"><code>R/filter.R</code></a></small>
      <div class="d-none name"><code>remove_low_count_genes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>TODO this function also transforms raw counts to CPM, but that should be a separate function before this step, before filter_counts function()</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">remove_low_count_genes</span><span class="op">(</span></span>
<span>  <span class="va">counts_dat</span>,</span>
<span>  <span class="va">sample_metadata</span>,</span>
<span>  <span class="va">feature_id_colname</span>,</span>
<span>  <span class="va">group_colname</span>,</span>
<span>  use_cpm_counts_to_filter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  use_group_based_filtering <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  minimum_count_value_to_be_considered_nonzero <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  minimum_number_of_samples_with_nonzero_counts_in_total <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  minimum_number_of_samples_with_nonzero_counts_in_a_group <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-feature-id-colname">feature_id_colname<a class="anchor" aria-label="anchor" href="#arg-feature-id-colname"></a></dt>
<dd><p>The column from the counts dataa containing the Feature IDs (Usually Gene or Protein ID). This is usually the first column of your input Counts Matrix. Only columns of Text type from your input Counts Matrix will be available to select for this parameter. (Default: <code>NULL</code> - first column in the counts matrix will be used.)</p></dd>


<dt id="arg-group-colname">group_colname<a class="anchor" aria-label="anchor" href="#arg-group-colname"></a></dt>
<dd><p>The column from the sample metadata containing the sample group information. This is usually a column showing to which experimental treatments each sample belongs (e.g. WildType, Knockout, Tumor, Normal, Before, After, etc.).</p></dd>


<dt id="arg-use-cpm-counts-to-filter">use_cpm_counts_to_filter<a class="anchor" aria-label="anchor" href="#arg-use-cpm-counts-to-filter"></a></dt>
<dd><p>If no transformation has been been performed on counts matrix (eg Raw Counts) set to TRUE. If TRUE counts will be transformed to CPM and filtered based on given criteria. If gene counts matrix has been transformed (eg log2, CPM, FPKM or some form of Normalization) set to FALSE. If FALSE no further transformation will be applied and features will be filtered as is. For RNAseq data RAW counts should be transformed to CPM in order to properly filter.</p></dd>


<dt id="arg-use-group-based-filtering">use_group_based_filtering<a class="anchor" aria-label="anchor" href="#arg-use-group-based-filtering"></a></dt>
<dd><p>If TRUE, only keeps features (e.g. genes) that have at least a certain number of samples with nonzero CPM counts in at least one group</p></dd>


<dt id="arg-minimum-count-value-to-be-considered-nonzero">minimum_count_value_to_be_considered_nonzero<a class="anchor" aria-label="anchor" href="#arg-minimum-count-value-to-be-considered-nonzero"></a></dt>
<dd><p>Minimum count value to be considered non-zero for a sample</p></dd>


<dt id="arg-minimum-number-of-samples-with-nonzero-counts-in-total">minimum_number_of_samples_with_nonzero_counts_in_total<a class="anchor" aria-label="anchor" href="#arg-minimum-number-of-samples-with-nonzero-counts-in-total"></a></dt>
<dd><p>Minimum number of samples (total) with non-zero counts</p></dd>


<dt id="arg-minimum-number-of-samples-with-nonzero-counts-in-a-group">minimum_number_of_samples_with_nonzero_counts_in_a_group<a class="anchor" aria-label="anchor" href="#arg-minimum-number-of-samples-with-nonzero-counts-in-a-group"></a></dt>
<dd><p>Only keeps genes that have at least this number of samples with nonzero CPM counts in at least one group</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>counts matrix with low-count genes removed</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

  </div></footer></body></html>

