<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ MOSuite</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta name="robots" content="noindex"></head><body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/cli.html">Calling MOSuite from the CLI</a></li>
    <li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index" id="container">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="multiomicdataset">multiOmicDataSet<a class="anchor" aria-label="anchor" href="#multiomicdataset"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="create_multiOmicDataSet_from_dataframes.html">create_multiOmicDataSet_from_dataframes()</a></code>

        </dt>
        <dd>Construct a multiOmicDataSet object from data frames</dd>
      </dl><dl><dt>

          <code><a href="create_multiOmicDataSet_from_files.html">create_multiOmicDataSet_from_files()</a></code>

        </dt>
        <dd>Construct a multiOmicDataSet object from tsv files.</dd>
      </dl><dl><dt>

          <code><a href="multiOmicDataSet.html">multiOmicDataSet()</a></code>

        </dt>
        <dd>multiOmicDataSet class</dd>
      </dl></div><div class="section level2">
      <h2 id="main-functions">Main functions<a class="anchor" aria-label="anchor" href="#main-functions"></a></h2>

      <div class="section-desc"><p>See <code><a href="../articles/intro.html">vignette('intro')</a></code> for recommended usage</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="batch_correct_counts.html">batch_correct_counts()</a></code>

        </dt>
        <dd>Perform batch correction</dd>
      </dl><dl><dt>

          <code><a href="clean_raw_counts.html">clean_raw_counts()</a></code>

        </dt>
        <dd>Clean Raw Counts</dd>
      </dl><dl><dt>

          <code><a href="diff_counts.html">diff_counts()</a></code>

        </dt>
        <dd>Differential expression analysis</dd>
      </dl><dl><dt>

          <code><a href="extract_counts.html">extract_counts()</a></code>

        </dt>
        <dd>Extract count data</dd>
      </dl><dl><dt>

          <code><a href="filter_counts.html">filter_counts()</a></code>

        </dt>
        <dd>Filter low counts</dd>
      </dl><dl><dt>

          <code><a href="normalize_counts.html">normalize_counts()</a></code>

        </dt>
        <dd>Normalize counts</dd>
      </dl><dl><dt>

          <code><a href="filter_diff.html">filter_diff()</a></code>

        </dt>
        <dd>Filter features from differential analysis based on statistical significance</dd>
      </dl></div><div class="section level2">
      <h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="plot_corr_heatmap.html">plot_corr_heatmap()</a></code>

        </dt>
        <dd>Plot correlation heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_expr_heatmap.html">plot_expr_heatmap()</a></code>

        </dt>
        <dd>Plot expression heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_histogram.html">plot_histogram()</a></code>

        </dt>
        <dd>Plot histogram</dd>
      </dl><dl><dt>

          <code><a href="plot_pca.html">plot_pca()</a></code>

        </dt>
        <dd>Perform and plot a Principal Components Analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_read_depth.html">plot_read_depth()</a></code>

        </dt>
        <dd>Plot read depth as a bar plot</dd>
      </dl><dl><dt>

          <code><a href="plot_venn_diagram.html">plot_venn_diagram()</a></code>

        </dt>
        <dd>Plot a venn diagram, UpSet plot, or table of intersections</dd>
      </dl><dl><dt>

          <code><a href="plot_volcano_enhanced.html">plot_volcano_enhanced()</a></code>

        </dt>
        <dd>Enhanced Volcano Plot</dd>
      </dl><dl><dt>

          <code><a href="plot_volcano_summary.html">plot_volcano_summary()</a></code>

        </dt>
        <dd>Volcano Plot - Summary [CC Produces one volcano plot for each tested contrast in the input DEG table.</dd>
      </dl><dl><dt>

          <code><a href="print_or_save_plot.html">print_or_save_plot()</a></code>

        </dt>
        <dd>Print and/or save a ggplot</dd>
      </dl></div><div class="section level2">

      <h3 id="color-selectors">Color selectors<a class="anchor" aria-label="anchor" href="#color-selectors"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="get_colors_lst.html">get_colors_lst()</a></code>

        </dt>
        <dd>Create named list of default colors for plotting</dd>
      </dl><dl><dt>

          <code><a href="get_colors_vctr.html">get_colors_vctr()</a></code>

        </dt>
        <dd>Get vector of colors for observations in one column of a data frame</dd>
      </dl><dl><dt>

          <code><a href="set_color_pal.html">set_color_pal()</a></code>

        </dt>
        <dd>Set color palette for a single group/column</dd>
      </dl></div><div class="section level2">
      <h2 id="misc-helpers-utilities">Misc. helpers &amp; utilities<a class="anchor" aria-label="anchor" href="#misc-helpers-utilities"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="calc_cpm.html">calc_cpm()</a></code>

        </dt>
        <dd>Calculate counts-per-million (CPM) on raw counts in a multiOmicDataSet</dd>
      </dl><dl><dt>

          <code><a href="calc_pca.html">calc_pca()</a></code>

        </dt>
        <dd>Perform principal components analysis</dd>
      </dl><dl><dt>

          <code><a href="bind_dfs_long.html">bind_dfs_long()</a></code>

        </dt>
        <dd>Bind dataframes in named list to long dataframe</dd>
      </dl><dl><dt>

          <code><a href="join_dfs_wide.html">join_dfs_wide()</a></code>

        </dt>
        <dd>Join dataframes in named list to wide dataframe</dd>
      </dl></div><div class="section level2">
      <h2 id="datasets">Datasets<a class="anchor" aria-label="anchor" href="#datasets"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="gene_counts.html">gene_counts</a></code>

        </dt>
        <dd>RSEM expected gene counts</dd>
      </dl><dl><dt>

          <code><a href="nidap_batch_corrected_counts.html">nidap_batch_corrected_counts</a></code>

        </dt>
        <dd>Batch-corrected counts for the NIDAP test dataset.</dd>
      </dl><dl><dt>

          <code><a href="nidap_batch_corrected_counts_2.html">nidap_batch_corrected_counts_2</a></code>

        </dt>
        <dd>Batch-corrected counts for the NIDAP test dataset. The result of running <code><a href="../reference/batch_correct_counts.html">batch_correct_counts()</a></code> on <code>nidap_norm_counts</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_clean_raw_counts.html">nidap_clean_raw_counts</a></code>

        </dt>
        <dd>Clean raw counts for the NIDAP test dataset. The result of running <code><a href="../reference/clean_raw_counts.html">clean_raw_counts()</a></code> on <code>nidap_raw_counts</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_deg_analysis.html">nidap_deg_analysis</a></code>

        </dt>
        <dd>Differential gene expression analysis for the NIDAP test dataset.</dd>
      </dl><dl><dt>

          <code><a href="nidap_deg_analysis_2.html">nidap_deg_analysis_2</a></code>

        </dt>
        <dd>Differential gene expression analysis for the NIDAP test dataset. The result of running <code><a href="../reference/diff_counts.html">diff_counts()</a></code> on <code>nidap_filtered_counts</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_deg_gene_list.html">nidap_deg_gene_list</a></code>

        </dt>
        <dd>List of differentially expressed genes from the NIDAP test dataset using default parameters with <code><a href="../reference/filter_diff.html">filter_diff()</a></code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_filtered_counts.html">nidap_filtered_counts</a></code>

        </dt>
        <dd>Filtered counts for the NIDAP test dataset. The result of running <code><a href="../reference/filter_counts.html">filter_counts()</a></code> on <code>nidap_clean_raw_counts</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_norm_counts.html">nidap_norm_counts</a></code>

        </dt>
        <dd>Normalized counts for the NIDAP test dataset. The result of running <code><a href="../reference/normalize_counts.html">normalize_counts()</a></code> on <code>nidap_filtered_counts</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_raw_counts.html">nidap_raw_counts</a></code>

        </dt>
        <dd>Raw counts for the NIDAP test dataset Pairs with <code>nidap_sample_metadata</code>.</dd>
      </dl><dl><dt>

          <code><a href="nidap_sample_metadata.html">nidap_sample_metadata</a></code>

        </dt>
        <dd>Sample metadata for the NIDAP test dataset</dd>
      </dl><dl><dt>

          <code><a href="nidap_venn_diagram_dat.html">nidap_venn_diagram_dat</a></code>

        </dt>
        <dd>Output data from venn diagram. The result of running <code><a href="../reference/plot_venn_diagram.html">plot_venn_diagram()</a></code> on <code>nidap_volcano_summary_dat</code></dd>
      </dl><dl><dt>

          <code><a href="nidap_volcano_summary_dat.html">nidap_volcano_summary_dat</a></code>

        </dt>
        <dd>Summarized differential expression analysis for input to venn diagram</dd>
      </dl></div><div class="section level2">
      <h2 id="options">Options<a class="anchor" aria-label="anchor" href="#options"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="options.html">options</a></code>

        </dt>
        <dd>MOSuite Options</dd>
      </dl></div><div class="section level2">

      <h3 id="low-level-plot-helpers">Low-level plot helpers<a class="anchor" aria-label="anchor" href="#low-level-plot-helpers"></a></h3>
      <div class="section-desc"><p>These functions are called by the main visualization functions.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="plot_corr_heatmap_dat.html">plot_corr_heatmap_dat</a></code>

        </dt>
        <dd>Plot correlation heatmap for counts dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_corr_heatmap_moo.html">plot_corr_heatmap_moo</a></code>

        </dt>
        <dd>Plot correlation heatmap for multiOmicDataSet</dd>
      </dl><dl><dt>

          <code><a href="plot_expr_heatmap_dat.html">plot_expr_heatmap_dat</a></code>

        </dt>
        <dd>Plot expression heatmap for counts dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_expr_heatmap_moo.html">plot_expr_heatmap_moo</a></code>

        </dt>
        <dd>Plot expression heatmap for multiOmicDataSet</dd>
      </dl><dl><dt>

          <code><a href="plot_histogram_dat.html">plot_histogram_dat</a></code>

        </dt>
        <dd>Plot histogram for counts dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_histogram_moo.html">plot_histogram_moo</a></code>

        </dt>
        <dd>Plot histogram for multiOmicDataSet</dd>
      </dl><dl><dt>

          <code><a href="plot_pca_2d.html">plot_pca_2d()</a></code>

        </dt>
        <dd>Perform and plot a 2D Principal Components Analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_pca_3d.html">plot_pca_3d()</a></code>

        </dt>
        <dd>3D PCA for counts dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_pca_dat.html">plot_pca_dat</a></code>

        </dt>
        <dd>Plot 2D or 3D PCA for counts dataframe</dd>
      </dl><dl><dt>

          <code><a href="plot_pca_moo.html">plot_pca_moo</a></code>

        </dt>
        <dd>Plot 2D or 3D PCA for multiOmicDataset</dd>
      </dl><dl><dt>

          <code><a href="plot_read_depth_dat.html">plot_read_depth_dat</a></code>

        </dt>
        <dd>Plot read depth for <code>data.frame</code></dd>
      </dl><dl><dt>

          <code><a href="plot_read_depth_moo.html">plot_read_depth_moo</a></code>

        </dt>
        <dd>Plot read depth for multiOmicDataSet</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer></body></html>

