<!DOCTYPE html>
<!-- source https://github.com/tidyverse/tidytemplate/blob/6345fdbee6f46cb39814f488addf793fb920c813/inst/pkgdown/BS5/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization with built-in plots â€¢ MOSuite</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Visualization with built-in plots">
</head>
<body>
    <a href="#container" class="visually-hidden-focusable">Skip to content</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MOSuite</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/intro.html">Introduction to MultiOmicsSuite</a></li>
    <li><a class="dropdown-item" href="../articles/renee.html">RSEM counts from RENEE</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization with built-in plots</a></li>
    <li><a class="dropdown-item" href="../articles/cli.html">Calling MOSuite from the CLI</a></li>
    <li><a class="dropdown-item" href="../articles/memory.html">Memory Usage</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CCBR/MOSuite/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article" id="container">



<link href="visualization_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="visualization_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="visualization_files/plotly-binding-4.11.0/plotly.js"></script><script src="visualization_files/typedarray-0.1/typedarray.min.js"></script><link href="visualization_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="visualization_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="visualization_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="visualization_files/plotly-main-2.11.1/plotly-latest.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualization with built-in plots</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CCBR/MOSuite/blob/v0.2.0/vignettes/visualization.Rmd" class="external-link"><code>vignettes/visualization.Rmd</code></a></small>
      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CCBR/MOSuite" class="external-link">MOSuite</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by</span></span>
<span><span class="co">#&gt; 'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</span></span></code></pre></div>
<div class="section level2">
<h2 id="default-plots-from-each-step">Default plots from each step<a class="anchor" aria-label="anchor" href="#default-plots-from-each-step"></a>
</h2>
<p>Default plots can be printed to the screen and/or saved to the
disk.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set options to print &amp; save the plots</span></span>
<span><span class="fu"><a href="../reference/options.html">options</a></span><span class="op">(</span>moo_print_plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/options.html">options</a></span><span class="op">(</span>moo_save_plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># when moo_save_plots is TRUE, plots are saved to this directory:</span></span>
<span><span class="fu"><a href="../reference/options.html">options</a></span><span class="op">(</span>moo_plots_dir <span class="op">=</span> <span class="st">"./figures"</span><span class="op">)</span></span></code></pre></div>
<p>See <code><a href="../reference/options.html">?MOSuite::options</a></code> for more information.</p>
<div class="section level3">
<h3 id="clean">clean<a class="anchor" aria-label="anchor" href="#clean"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_multiOmicDataSet_from_dataframes.html">create_multiOmicDataSet_from_dataframes</a></span><span class="op">(</span></span>
<span>  sample_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_sample_metadata</span><span class="op">)</span>,</span>
<span>  counts_dat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">nidap_raw_counts</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/clean_raw_counts.html">clean_raw_counts</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/nidap_data_clean-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span>
<span><span class="co">#&gt; * cleaning raw counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Not able to identify multiple id's in GeneName</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Columns that can be used to aggregate gene information GeneName</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Aggregating the counts for the same ID in different chromosome locations.</span></span>
<span><span class="co">#&gt; Column used to Aggregate duplicate IDs: GeneName</span></span>
<span><span class="co">#&gt; Number of rows before Collapse: 43280</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; no duplicated IDs in GeneName</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/filter_counts.html">filter_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span><span class="co">#&gt; * filtering clean counts</span></span>
<span><span class="co">#&gt; Number of features after filtering: 7943</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/nidap_filter-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_filter-2.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_filter-3.png" width="480"></p>
</div>
<div class="section level3">
<h3 id="normalize">normalize<a class="anchor" aria-label="anchor" href="#normalize"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize_counts.html">normalize_counts</a></span><span class="op">(</span>group_colname <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span><span class="co">#&gt; * normalizing filt counts</span></span>
<span><span class="co">#&gt; Total number of features included: 7943</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/nidap_norm-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_norm-2.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_norm-3.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Sample columns: A1, Sample columns: A2, Sample columns: A3, Sample columns: B1, Sample columns: B2, Sample columns: B3, Sample columns: C1, Sample columns: C2, Sample columns: C3</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="batch-correct">batch correct<a class="anchor" aria-label="anchor" href="#batch-correct"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/batch_correct_counts.html">batch_correct_counts</a></span><span class="op">(</span></span>
<span>    covariates_colname <span class="op">=</span> <span class="st">"Group"</span>,</span>
<span>    batch_colname <span class="op">=</span> <span class="st">"Batch"</span>,</span>
<span>    label_colname <span class="op">=</span> <span class="st">"Label"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; * batch-correcting norm-voom counts</span></span>
<span><span class="co">#&gt; Found2batches</span></span>
<span><span class="co">#&gt; Adjusting for2covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; Finding parametric adjustments</span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/nidap_batch-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_batch-2.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span></code></pre>
<p><img src="visualization_files/figure-html/nidap_batch-3.png" width="480"></p>
<pre><code><span><span class="co">#&gt; The total number of features in output: 7943</span></span>
<span><span class="co">#&gt; Number of samples after batch correction: 10</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="differential-expression">differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/diff_counts.html">diff_counts</a></span><span class="op">(</span></span>
<span>    count_type <span class="op">=</span> <span class="st">"filt"</span>,</span>
<span>    covariates_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="st">"Batch"</span><span class="op">)</span>,</span>
<span>    contrast_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group"</span><span class="op">)</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B-A"</span>, <span class="st">"C-A"</span>, <span class="st">"B-C"</span><span class="op">)</span>,</span>
<span>    input_in_log_counts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_mean_and_sd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    voom_normalization_method <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; * differential counts</span></span>
<span><span class="co">#&gt; Setting first column of `counts` as gene annotation.</span></span>
<span><span class="co">#&gt; Total number of genes included: 7943</span></span>
<span><span class="co">#&gt; `geom_smooth()` using method = 'gam' and formula = 'y ~ s(x, bs = "cs")'</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/diff_counts-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span>
<span><span class="co">#&gt; `geom_smooth()` using method = 'gam' and formula = 'y ~ s(x, bs = "cs")'</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="filter-differential-features">filter differential features<a class="anchor" aria-label="anchor" href="#filter-differential-features"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">moo</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html" class="external-link">%&lt;&gt;%</a></span> <span class="fu"><a href="../reference/filter_diff.html">filter_diff</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; * filtering differential features</span></span>
<span><span class="co">#&gt; Total number of genes selected with adjpval &lt; 0.05 and | logFC | â‰¥ 1 is</span></span>
<span><span class="co">#&gt; sum(selgenes)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/filter_diff-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="customize-plots">Customize plots<a class="anchor" aria-label="anchor" href="#customize-plots"></a>
</h2>
<p>TODO</p>
<ul>
<li>show how to use individual plotting functions</li>
<li>how to customize &amp; override default color palettes</li>
<li>how to customize ggplot objects</li>
</ul>
<div class="section level3">
<h3 id="d-pca">3D PCA<a class="anchor" aria-label="anchor" href="#d-pca"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">moo</span><span class="op">@</span><span class="va">counts</span><span class="op">$</span><span class="va">batch</span>,</span>
<span>  <span class="va">moo</span><span class="op">@</span><span class="va">sample_meta</span>,</span>
<span>  principal_components <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  group_colname <span class="op">=</span> <span class="st">"Group"</span>,</span>
<span>  label_colname <span class="op">=</span> <span class="st">"Label"</span>,</span>
<span>  color_values <span class="op">=</span> <span class="va">moo</span><span class="op">@</span><span class="va">analyses</span><span class="op">[[</span><span class="st">"colors"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Group"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:480px;height:384px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"visdat":{"2f09258bb54a":["function () ","plotlyVisDat"]},"cur_data":"2f09258bb54a","attrs":{"2f09258bb54a":{"x":{},"y":{},"z":{},"mode":"markers","marker":{"size":8},"hoverinfo":"text","text":{},"color":{},"size":24,"colors":["#000000","#E69F00","#56B4E9"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"xaxis":{"title":"PC1"},"yaxis":{"title":"PC2"},"zaxis":{"title":"PC3"}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[-37.02988740420713,-46.202726529239996,-47.388329488061188],"y":[33.20056607030736,40.838627625724037,28.814546147005224],"z":[4.7406847243841019,-10.906448043253027,-13.204218047115619],"mode":"markers","marker":{"color":"rgba(0,0,0,1)","size":8,"sizemode":"area","line":{"color":"rgba(0,0,0,1)"}},"hoverinfo":["text","text","text"],"text":["A1","A2","A3"],"type":"scatter3d","name":"A","textfont":{"color":"rgba(0,0,0,1)","size":55},"error_y":{"color":"rgba(0,0,0,1)","width":55},"error_x":{"color":"rgba(0,0,0,1)","width":55},"line":{"color":"rgba(0,0,0,1)","width":55},"frame":null},{"x":[-28.694926741752916,-13.178325203869875,-15.174264901109913],"y":[-29.61351670343015,-34.916643374714177,-65.245132469700494],"z":[25.172778558047199,13.334101581276443,7.1777565898972835],"mode":"markers","marker":{"color":"rgba(230,159,0,1)","size":8,"sizemode":"area","line":{"color":"rgba(230,159,0,1)"}},"hoverinfo":["text","text","text"],"text":["B1","B2","B3"],"type":"scatter3d","name":"B","textfont":{"color":"rgba(230,159,0,1)","size":55},"error_y":{"color":"rgba(230,159,0,1)","width":55},"error_x":{"color":"rgba(230,159,0,1)","width":55},"line":{"color":"rgba(230,159,0,1)","width":55},"frame":null},{"x":[54.383108042646533,77.413153148631352,55.872199076963149],"y":[-0.31296701193061999,31.436049026500761,-4.2015293097619741],"z":[-44.493203749203325,67.633314692756926,-49.454766306789963],"mode":"markers","marker":{"color":"rgba(86,180,233,1)","size":8,"sizemode":"area","line":{"color":"rgba(86,180,233,1)"}},"hoverinfo":["text","text","text"],"text":["C1","C2","C3"],"type":"scatter3d","name":"C","textfont":{"color":"rgba(86,180,233,1)","size":55},"error_y":{"color":"rgba(86,180,233,1)","width":55},"error_x":{"color":"rgba(86,180,233,1)","width":55},"line":{"color":"rgba(86,180,233,1)","width":55},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="expression-heatmap">Expression Heatmap<a class="anchor" aria-label="anchor" href="#expression-heatmap"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">heatmap_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_expr_heatmap.html">plot_expr_heatmap</a></span><span class="op">(</span><span class="va">moo</span>, count_type <span class="op">=</span> <span class="st">"norm"</span>, sub_count_type <span class="op">=</span> <span class="st">"voom"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `arrange_()` was deprecated in dplyr 0.7.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Please use `arrange()` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> See vignette('programming') for more help</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> The deprecated feature was likely used in the <span style="color: #0000BB;">MOSuite</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/CCBR/MOSuite/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; [1] "The total number of genes in heatmap: 500"</span></span>
<span><span class="co">#&gt; Warning: The input is a data frame, convert it to the matrix.</span></span>
<span><span class="co">#&gt; Warning: argument `height` is not supported in pheatmap -&gt; Heatmap translation,</span></span>
<span><span class="co">#&gt; skip it.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">heatmap_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/expr_heatmap-1.png" width="480"></p>
</div>
<div class="section level3">
<h3 id="volcano">Volcano<a class="anchor" aria-label="anchor" href="#volcano"></a>
</h3>
<p>= #### Summary</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_volcano_summary</span> <span class="op">&lt;-</span> <span class="va">moo</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/join_dfs_wide.html">join_dfs_wide</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_volcano_summary.html">plot_volcano_summary</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Preparing table for contrast: B-A</span></span>
<span><span class="co">#&gt; Fold change column: B-A_logFC</span></span>
<span><span class="co">#&gt; pval column: B-A_pval</span></span>
<span><span class="co">#&gt; Total number of features included in volcano plot: 7943</span></span>
<span><span class="co">#&gt; Preparing table for contrast: C-A</span></span>
<span><span class="co">#&gt; Fold change column: C-A_logFC</span></span>
<span><span class="co">#&gt; pval column: C-A_pval</span></span>
<span><span class="co">#&gt; Total number of features included in volcano plot: 7943</span></span>
<span><span class="co">#&gt; Preparing table for contrast: B-C</span></span>
<span><span class="co">#&gt; Fold change column: B-C_logFC</span></span>
<span><span class="co">#&gt; pval column: B-C_pval</span></span>
<span><span class="co">#&gt; Total number of features included in volcano plot: 7943</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 27 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 29 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 29 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/volcano_summary-1.png" width="480"></p>
<pre><code><span><span class="co">#&gt; Saving 5 x 4 in image</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 27 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 29 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 29 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dat_volcano_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt;       GeneName Contrast         FC     logFC     tstat         pval</span></span>
<span><span class="co">#&gt; B-A.1     Dntt      B-A -42.746586 -5.417737 -15.68797 3.159343e-09</span></span>
<span><span class="co">#&gt; B-A.2   Tmsb4x      B-A   3.850020  1.944866  12.91026 2.760555e-08</span></span>
<span><span class="co">#&gt; B-A.3     Flt3      B-A  -7.714394 -2.947553 -11.38084 1.093405e-07</span></span>
<span><span class="co">#&gt; B-A.4  Tspan13      B-A  -7.038498 -2.815268 -11.03127 1.531110e-07</span></span>
<span><span class="co">#&gt; B-A.5    Tapt1      B-A  -5.291816 -2.403763 -10.65847 2.214593e-07</span></span>
<span><span class="co">#&gt; B-A.6    Itgb7      B-A   8.873823  3.149556  10.56147 2.442070e-07</span></span>
<span><span class="co">#&gt;            adjpval</span></span>
<span><span class="co">#&gt; B-A.1 2.509467e-05</span></span>
<span><span class="co">#&gt; B-A.2 1.096354e-04</span></span>
<span><span class="co">#&gt; B-A.3 2.894973e-04</span></span>
<span><span class="co">#&gt; B-A.4 3.040401e-04</span></span>
<span><span class="co">#&gt; B-A.5 3.232894e-04</span></span>
<span><span class="co">#&gt; B-A.6 3.232894e-04</span></span></code></pre>
<div class="section level4">
<h4 id="enhanced">Enhanced<a class="anchor" aria-label="anchor" href="#enhanced"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_volcano_enhanced</span> <span class="op">&lt;-</span> <span class="va">moo</span><span class="op">@</span><span class="va">analyses</span><span class="op">$</span><span class="va">diff</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/join_dfs_wide.html">join_dfs_wide</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_volcano_enhanced.html">plot_volcano_enhanced</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(GeneName)`</span></span>
<span><span class="co">#&gt; Genes in initial dataset: 7943</span></span>
<span><span class="co">#&gt; Max y: 4.60041859457819</span></span>
<span><span class="co">#&gt; Genes in initial dataset: 7943</span></span>
<span><span class="co">#&gt; Max y: 4.32577808863472</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/volcano_enhanced-1.png" width="480"></p>
</div>
</div>
<div class="section level3">
<h3 id="venn-diagram">Venn Diagram<a class="anchor" aria-label="anchor" href="#venn-diagram"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">venn_dat</span> <span class="op">&lt;-</span> <span class="va">dat_volcano_summary</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_venn_diagram.html">plot_venn_diagram</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; All intersections: 1:7,c(1, 2, 3, 4, 5, 6, 7),c(79, 119, 265, 493, 149, 271, 516),c("Yes", "Yes", "Yes", "Yes", "Yes", "Yes", "Yes")</span></span>
<span><span class="co">#&gt; Intersections returned: 1:7,c(1, 2, 3, 4, 5, 6, 7),c(79, 119, 265, 493, 149, 271, 516)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/venn_diagram-1.png" width="480"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">venn_dat</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Gene      Intersection Id Size</span></span>
<span><span class="co">#&gt; 1    1 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span>
<span><span class="co">#&gt; 2    3 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span>
<span><span class="co">#&gt; 3   11 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span>
<span><span class="co">#&gt; 4   13 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span>
<span><span class="co">#&gt; 5   31 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span>
<span><span class="co">#&gt; 6   34 (B-A âˆ© B-C âˆ© C-A)  1   79</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



   </div>
  <footer><div class="container">
  <div class="pkgdown-footer-left">
  <p>Created by the <a href="https://github.com/CCBR" class="external-link">CCR Collaborative Bioinformatics Resource</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

  </div></footer>
</body>
</html>
